*** ./smali/android/provider/Settings$System.smali	2014-08-20 05:48:02.496013341 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/framework.jar.out/./smali/android/provider/Settings$System.smali	2014-08-20 05:50:47.344017615 +0700
***************
*** 140,145 ****
--- 140,147 ----
  
  .field public static final EGG_MODE:Ljava/lang/String; = "egg_mode"
  
+ .field public static final EMERGENCY_TONE:Ljava/lang/String; = "emergency_tone"
+ 
  .field public static final END_BUTTON_BEHAVIOR:Ljava/lang/String; = "end_button_behavior"
  
  .field public static final END_BUTTON_BEHAVIOR_DEFAULT:I = 0x2
***************
*** 2475,2480 ****
--- 2477,2486 ----
  
      invoke-static {v1, v2}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     invoke-static {p0, p1, p2, p3}, Landroid/provider/Settings$Injector;->putSecureStringForUser(Landroid/content/ContentResolver;Ljava/lang/String;Ljava/lang/String;I)Z
+ 
+     move-result v0
+ 
      :goto_0
      return v0
  
***************
*** 2530,2535 ****
--- 2536,2545 ----
  
      invoke-static {v1, v2}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     invoke-static {p0, p1, p2, p3}, Landroid/provider/Settings$Injector;->putGlobalStringForUser(Landroid/content/ContentResolver;Ljava/lang/String;Ljava/lang/String;I)Z
+ 
+     move-result v0
+ 
      goto :goto_0
  
      :cond_2
