*** ./smali/com/android/server/power/DisplayPowerController.smali	2014-08-20 05:49:38.552015832 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/services.jar.out/./smali/com/android/server/power/DisplayPowerController.smali	2014-08-20 05:52:23.848020118 +0700
***************
*** 31,36 ****
--- 31,40 ----
  
  .field private static final DEBUG_PRETEND_PROXIMITY_SENSOR_ABSENT:Z = false
  
+ .field private static final DEFALUT_SCREEN_COLOR:I = 0x2
+ 
+ .field private static final DEFAULT_SCREEN_SATURATION:I = 0xb
+ 
  .field private static final ELECTRON_BEAM_OFF_ANIMATION_DURATION_MILLIS:I = 0x190
  
  .field private static final ELECTRON_BEAM_ON_ANIMATION_DURATION_MILLIS:I = 0xfa
***************
*** 45,50 ****
--- 49,58 ----
  
  .field private static final MSG_UPDATE_POWER_STATE:I = 0x1
  
+ .field private static final PROPERTY_SCREEN_COLOR:Ljava/lang/String; = "persist.sys.display_prefer"
+ 
+ .field private static final PROPERTY_SCREEN_SATURATION:Ljava/lang/String; = "persist.sys.display_ce"
+ 
  .field private static final PROXIMITY_NEGATIVE:I = 0x0
  
  .field private static final PROXIMITY_POSITIVE:I = 0x1
***************
*** 2571,2576 ****
--- 2579,2694 ----
      goto :goto_0
  .end method
  
+ .method public static setScreenColor()V
+     .locals 6
+ 
+     .prologue
+     const/16 v5, 0xb
+ 
+     const/4 v4, 0x2
+ 
+     const-string v1, "persist.sys.display_prefer"
+ 
+     invoke-static {v1, v4}, Landroid/os/SystemProperties;->getInt(Ljava/lang/String;I)I
+ 
+     move-result v0
+ 
+     .local v0, currValue:I
+     const-string v1, "DisplayPowerController"
+ 
+     new-instance v2, Ljava/lang/StringBuilder;
+ 
+     invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
+ 
+     const-string v3, "PROPERTY_SCREEN_COLOR set, currValue:"
+ 
+     invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v2
+ 
+     invoke-virtual {v2, v0}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
+ 
+     move-result-object v2
+ 
+     const-string v3, ", default:"
+ 
+     invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v2
+ 
+     invoke-virtual {v2, v4}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
+ 
+     move-result-object v2
+ 
+     invoke-virtual {v2}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
+ 
+     move-result-object v2
+ 
+     invoke-static {v1, v2}, Landroid/util/Slog;->d(Ljava/lang/String;Ljava/lang/String;)I
+ 
+     if-eq v0, v4, :cond_0
+ 
+     const-string v1, "persist.sys.display_prefer"
+ 
+     invoke-static {v0}, Ljava/lang/String;->valueOf(I)Ljava/lang/String;
+ 
+     move-result-object v2
+ 
+     invoke-static {v1, v2}, Landroid/os/SystemProperties;->set(Ljava/lang/String;Ljava/lang/String;)V
+ 
+     :cond_0
+     const-string v1, "persist.sys.display_ce"
+ 
+     invoke-static {v1, v5}, Landroid/os/SystemProperties;->getInt(Ljava/lang/String;I)I
+ 
+     move-result v0
+ 
+     const-string v1, "DisplayPowerController"
+ 
+     new-instance v2, Ljava/lang/StringBuilder;
+ 
+     invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
+ 
+     const-string v3, "PROPERTY_SCREEN_SATURATION set, currValue:"
+ 
+     invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v2
+ 
+     invoke-virtual {v2, v0}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
+ 
+     move-result-object v2
+ 
+     const-string v3, ", default:"
+ 
+     invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v2
+ 
+     invoke-virtual {v2, v5}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
+ 
+     move-result-object v2
+ 
+     invoke-virtual {v2}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
+ 
+     move-result-object v2
+ 
+     invoke-static {v1, v2}, Landroid/util/Slog;->d(Ljava/lang/String;Ljava/lang/String;)I
+ 
+     if-eq v0, v5, :cond_1
+ 
+     const-string v1, "persist.sys.display_ce"
+ 
+     invoke-static {v0}, Ljava/lang/String;->valueOf(I)Ljava/lang/String;
+ 
+     move-result-object v2
+ 
+     invoke-static {v1, v2}, Landroid/os/SystemProperties;->set(Ljava/lang/String;Ljava/lang/String;)V
+ 
+     :cond_1
+     return-void
+ .end method
+ 
  .method private setScreenOn(Z)V
      .locals 1
      .parameter "on"
***************
*** 2599,2606 ****
  
      invoke-virtual {v0}, Lcom/android/server/power/Notifier;->onScreenOn()V
  
-     :cond_0
      :goto_1
      return-void
  
      :cond_1
--- 2717,2726 ----
  
      invoke-virtual {v0}, Lcom/android/server/power/Notifier;->onScreenOn()V
  
      :goto_1
+     invoke-static {p1}, Lcom/android/server/power/PowerManagerService$Injector;->setScreenOn(Z)V
+ 
+     :cond_0
      return-void
  
      :cond_1
