*** ./smali/com/android/internal/widget/ActionBarContextView.smali	2014-08-20 05:49:27.124015535 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/framework2.jar.out/./smali/com/android/internal/widget/ActionBarContextView.smali	2014-08-20 05:52:12.308019819 +0700
***************
*** 162,167 ****
--- 162,176 ----
  
      const/4 v5, 0x0
  
+     invoke-virtual {p0}, Lcom/android/internal/widget/ActionBarContextView;->miuiInitTitle()Z
+ 
+     move-result v3
+ 
+     if-eqz v3, :cond_miui
+ 
+     return-void
+ 
+     :cond_miui
      iget-object v3, p0, Lcom/android/internal/widget/ActionBarContextView;->mTitleLayout:Landroid/widget/LinearLayout;
  
      if-nez v3, :cond_1
***************
*** 677,682 ****
--- 686,700 ----
      return-object v0
  .end method
  
+ .method protected getAnimationMode()I
+     .locals 1
+ 
+     .prologue
+     iget v0, p0, Lcom/android/internal/widget/ActionBarContextView;->mAnimationMode:I
+ 
+     return v0
+ .end method
+ 
  .method public getSubtitle()Ljava/lang/CharSequence;
      .locals 1
  
***************
*** 695,700 ****
--- 713,745 ----
      return-object v0
  .end method
  
+ .method public getTitleLayout()Landroid/widget/LinearLayout;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/widget/ActionBarContextView;->mTitleLayout:Landroid/widget/LinearLayout;
+ 
+     return-object v0
+ .end method
+ 
+ .method public getTitleStyleRes()I
+     .locals 1
+ 
+     .prologue
+     iget v0, p0, Lcom/android/internal/widget/ActionBarContextView;->mTitleStyleRes:I
+ 
+     return v0
+ .end method
+ 
+ .method public getTitleView()Landroid/widget/TextView;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/widget/ActionBarContextView;->mTitleView:Landroid/widget/TextView;
+ 
+     return-object v0
+ .end method
+ 
  .method public hideOverflowMenu()Z
      .locals 1
  
***************
*** 983,988 ****
--- 1028,1042 ----
      return-void
  .end method
  
+ .method protected miuiInitTitle()Z
+     .locals 1
+ 
+     .prologue
+     const/4 v0, 0x0
+ 
+     return v0
+ .end method
+ 
  .method public onAnimationCancel(Landroid/animation/Animator;)V
      .locals 0
      .parameter "animation"
***************
*** 1961,1966 ****
--- 2015,2030 ----
      goto :goto_9
  .end method
  
+ .method protected setAnimationMode(I)V
+     .locals 0
+     .parameter "mode"
+ 
+     .prologue
+     iput p1, p0, Lcom/android/internal/widget/ActionBarContextView;->mAnimationMode:I
+ 
+     return-void
+ .end method
+ 
  .method public setContentHeight(I)V
      .locals 0
      .parameter "height"
***************
*** 2178,2183 ****
--- 2242,2257 ----
      return-void
  .end method
  
+ .method public setTitleLayout(Landroid/widget/LinearLayout;)V
+     .locals 0
+     .parameter "titleLayout"
+ 
+     .prologue
+     iput-object p1, p0, Lcom/android/internal/widget/ActionBarContextView;->mTitleLayout:Landroid/widget/LinearLayout;
+ 
+     return-void
+ .end method
+ 
  .method public setTitleOptional(Z)V
      .locals 1
      .parameter "titleOptional"
***************
*** 2194,2199 ****
--- 2268,2283 ----
  
      return-void
  .end method
+ 
+ .method public setTitleView(Landroid/widget/TextView;)V
+     .locals 0
+     .parameter "titleView"
+ 
+     .prologue
+     iput-object p1, p0, Lcom/android/internal/widget/ActionBarContextView;->mTitleView:Landroid/widget/TextView;
+ 
+     return-void
+ .end method
  
  .method public shouldDelayChildPressedState()Z
      .locals 1
