*** ./smali/com/android/internal/widget/ActionBarView.smali	2014-08-20 05:49:25.860015503 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/framework2.jar.out/./smali/com/android/internal/widget/ActionBarView.smali	2014-08-20 05:52:11.020019785 +0700
***************
*** 68,73 ****
--- 68,75 ----
  
  .field private mMenuPrepared:Z
  
+ .field mMenuPresenterCallback:Lcom/android/internal/view/menu/MenuPresenter$Callback;
+ 
  .field private final mNavItemSelectedListener:Landroid/widget/AdapterView$OnItemSelectedListener;
  
  .field private mNavigationMode:I
***************
*** 96,105 ****
  
  .field private mTitle:Ljava/lang/CharSequence;
  
! .field private mTitleLayout:Landroid/widget/LinearLayout;
  
  .field private mTitleStyleRes:I
  
  .field private mTitleView:Landroid/widget/TextView;
  
  .field private final mUpClickListener:Landroid/view/View$OnClickListener;
--- 98,109 ----
  
  .field private mTitle:Ljava/lang/CharSequence;
  
! .field private mTitleLayout:Landroid/view/ViewGroup;
  
  .field private mTitleStyleRes:I
  
+ .field private mTitleUpView:Landroid/view/View;
+ 
  .field private mTitleView:Landroid/widget/TextView;
  
  .field private final mUpClickListener:Landroid/view/View$OnClickListener;
***************
*** 601,612 ****
      return-object v0
  .end method
  
! .method static synthetic access$1000(Lcom/android/internal/widget/ActionBarView;)Landroid/widget/LinearLayout;
      .locals 1
      .parameter "x0"
  
      .prologue
!     iget-object v0, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/widget/LinearLayout;
  
      return-object v0
  .end method
--- 605,616 ----
      return-object v0
  .end method
  
! .method static synthetic access$1000(Lcom/android/internal/widget/ActionBarView;)Landroid/view/ViewGroup;
      .locals 1
      .parameter "x0"
  
      .prologue
!     iget-object v0, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/view/ViewGroup;
  
      return-object v0
  .end method
***************
*** 929,937 ****
      .prologue
      const/4 v4, 0x0
  
!     iget-object v1, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/widget/LinearLayout;
  
!     if-nez v1, :cond_3
  
      invoke-virtual {p0}, Lcom/android/internal/widget/ActionBarView;->getContext()Landroid/content/Context;
  
--- 933,966 ----
      .prologue
      const/4 v4, 0x0
  
!     invoke-virtual {p0}, Lcom/android/internal/widget/ActionBarView;->miuiInitTitle()Z
  
!     move-result v1
! 
!     if-eqz v1, :cond_miui
! 
!     iget-object v1, p0, Lcom/android/internal/widget/ActionBarView;->mUpGoerFive:Landroid/view/ViewGroup;
! 
!     iget-object v2, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/view/ViewGroup;
! 
!     invoke-virtual {v1, v2}, Landroid/view/ViewGroup;->addView(Landroid/view/View;)V
! 
!     iget-object v1, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/view/ViewGroup;
! 
!     const v2, 0x60b00af
! 
!     invoke-virtual {v1, v2}, Landroid/view/ViewGroup;->findViewById(I)Landroid/view/View;
! 
!     move-result-object v1
! 
!     iput-object v1, p0, Lcom/android/internal/widget/ActionBarView;->mTitleUpView:Landroid/view/View;
! 
!     return-void
! 
!     :cond_miui
!     iget-object v1, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/view/ViewGroup;
! 
!     if-nez v1, :cond_4
  
      invoke-virtual {p0}, Lcom/android/internal/widget/ActionBarView;->getContext()Landroid/content/Context;
  
***************
*** 950,962 ****
  
      check-cast v1, Landroid/widget/LinearLayout;
  
!     iput-object v1, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/widget/LinearLayout;
  
!     iget-object v1, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/widget/LinearLayout;
  
      const v2, 0x1020265
  
!     invoke-virtual {v1, v2}, Landroid/widget/LinearLayout;->findViewById(I)Landroid/view/View;
  
      move-result-object v1
  
--- 979,991 ----
  
      check-cast v1, Landroid/widget/LinearLayout;
  
!     iput-object v1, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/view/ViewGroup;
  
!     iget-object v1, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/view/ViewGroup;
  
      const v2, 0x1020265
  
!     invoke-virtual {v1, v2}, Landroid/view/ViewGroup;->findViewById(I)Landroid/view/View;
  
      move-result-object v1
  
***************
*** 964,974 ****
  
      iput-object v1, p0, Lcom/android/internal/widget/ActionBarView;->mTitleView:Landroid/widget/TextView;
  
!     iget-object v1, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/widget/LinearLayout;
  
      const v2, 0x1020266
  
!     invoke-virtual {v1, v2}, Landroid/widget/LinearLayout;->findViewById(I)Landroid/view/View;
  
      move-result-object v1
  
--- 993,1003 ----
  
      iput-object v1, p0, Lcom/android/internal/widget/ActionBarView;->mTitleView:Landroid/widget/TextView;
  
!     iget-object v1, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/view/ViewGroup;
  
      const v2, 0x1020266
  
!     invoke-virtual {v1, v2}, Landroid/view/ViewGroup;->findViewById(I)Landroid/view/View;
  
      move-result-object v1
  
***************
*** 1033,1039 ****
  
      iget-object v1, p0, Lcom/android/internal/widget/ActionBarView;->mUpGoerFive:Landroid/view/ViewGroup;
  
!     iget-object v2, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/widget/LinearLayout;
  
      invoke-virtual {v1, v2}, Landroid/view/ViewGroup;->addView(Landroid/view/View;)V
  
--- 1062,1068 ----
  
      iget-object v1, p0, Lcom/android/internal/widget/ActionBarView;->mUpGoerFive:Landroid/view/ViewGroup;
  
!     iget-object v2, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/view/ViewGroup;
  
      invoke-virtual {v1, v2}, Landroid/view/ViewGroup;->addView(Landroid/view/View;)V
  
***************
*** 1058,1076 ****
      if-eqz v1, :cond_5
  
      :cond_4
!     iget-object v1, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/widget/LinearLayout;
  
      const/16 v2, 0x8
  
!     invoke-virtual {v1, v2}, Landroid/widget/LinearLayout;->setVisibility(I)V
  
      :goto_0
      return-void
  
      :cond_5
!     iget-object v1, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/widget/LinearLayout;
  
!     invoke-virtual {v1, v4}, Landroid/widget/LinearLayout;->setVisibility(I)V
  
      goto :goto_0
  .end method
--- 1087,1105 ----
      if-eqz v1, :cond_5
  
      :cond_4
!     iget-object v1, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/view/ViewGroup;
  
      const/16 v2, 0x8
  
!     invoke-virtual {v1, v2}, Landroid/view/ViewGroup;->setVisibility(I)V
  
      :goto_0
      return-void
  
      :cond_5
!     iget-object v1, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/view/ViewGroup;
  
!     invoke-virtual {v1, v4}, Landroid/view/ViewGroup;->setVisibility(I)V
  
      goto :goto_0
  .end method
***************
*** 1157,1168 ****
  
      .local v0, visible:Z
      :goto_0
!     iget-object v2, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/widget/LinearLayout;
  
      if-eqz v0, :cond_4
  
      :goto_1
!     invoke-virtual {v2, v1}, Landroid/widget/LinearLayout;->setVisibility(I)V
  
      .end local v0           #visible:Z
      :cond_1
--- 1186,1197 ----
  
      .local v0, visible:Z
      :goto_0
!     iget-object v2, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/view/ViewGroup;
  
      if-eqz v0, :cond_4
  
      :goto_1
!     invoke-virtual {v2, v1}, Landroid/view/ViewGroup;->setVisibility(I)V
  
      .end local v0           #visible:Z
      :cond_1
***************
*** 1351,1356 ****
--- 1380,1394 ----
      return v0
  .end method
  
+ .method public getExpandedActionView()Landroid/view/View;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/widget/ActionBarView;->mExpandedActionView:Landroid/view/View;
+ 
+     return-object v0
+ .end method
+ 
  .method public getNavigationMode()I
      .locals 1
  
***************
*** 1369,1374 ****
--- 1407,1439 ----
      return-object v0
  .end method
  
+ .method public getSubtitleStyleRes()I
+     .locals 1
+ 
+     .prologue
+     iget v0, p0, Lcom/android/internal/widget/ActionBarView;->mSubtitleStyleRes:I
+ 
+     return v0
+ .end method
+ 
+ .method public getSubtitleView()Landroid/widget/TextView;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/widget/ActionBarView;->mSubtitleView:Landroid/widget/TextView;
+ 
+     return-object v0
+ .end method
+ 
+ .method public getTabScrollView()Lcom/android/internal/widget/ScrollingTabContainerView;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/widget/ActionBarView;->mTabScrollView:Lcom/android/internal/widget/ScrollingTabContainerView;
+ 
+     return-object v0
+ .end method
+ 
  .method public getTitle()Ljava/lang/CharSequence;
      .locals 1
  
***************
*** 1378,1383 ****
--- 1443,1484 ----
      return-object v0
  .end method
  
+ .method public getTitleLayout()Landroid/view/ViewGroup;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/view/ViewGroup;
+ 
+     return-object v0
+ .end method
+ 
+ .method public getTitleStyleRes()I
+     .locals 1
+ 
+     .prologue
+     iget v0, p0, Lcom/android/internal/widget/ActionBarView;->mTitleStyleRes:I
+ 
+     return v0
+ .end method
+ 
+ .method public getTitleView()Landroid/widget/TextView;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/widget/ActionBarView;->mTitleView:Landroid/widget/TextView;
+ 
+     return-object v0
+ .end method
+ 
+ .method public getUpClickListener()Landroid/view/View$OnClickListener;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/widget/ActionBarView;->mUpClickListener:Landroid/view/View$OnClickListener;
+ 
+     return-object v0
+ .end method
+ 
  .method public hasEmbeddedTabs()Z
      .locals 1
  
***************
*** 1599,1604 ****
--- 1700,1718 ----
      goto :goto_1
  .end method
  
+ .method protected miuiInitTitle()Z
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/widget/ActionBarView;->mContext:Landroid/content/Context;
+ 
+     invoke-static {v0}, Lmiui/util/UiUtils;->isV5Ui(Landroid/content/Context;)Z
+ 
+     move-result v0
+ 
+     return v0
+ .end method
+ 
  .method protected onConfigurationChanged(Landroid/content/res/Configuration;)V
      .locals 4
      .parameter "newConfig"
***************
*** 1612,1624 ****
  
      iput-object v3, p0, Lcom/android/internal/widget/ActionBarView;->mSubtitleView:Landroid/widget/TextView;
  
!     iget-object v1, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/widget/LinearLayout;
  
      if-eqz v1, :cond_0
  
!     iget-object v1, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/widget/LinearLayout;
  
!     invoke-virtual {v1}, Landroid/widget/LinearLayout;->getParent()Landroid/view/ViewParent;
  
      move-result-object v1
  
--- 1726,1740 ----
  
      iput-object v3, p0, Lcom/android/internal/widget/ActionBarView;->mSubtitleView:Landroid/widget/TextView;
  
!     iput-object v3, p0, Lcom/android/internal/widget/ActionBarView;->mTitleUpView:Landroid/view/View;
! 
!     iget-object v1, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/view/ViewGroup;
  
      if-eqz v1, :cond_0
  
!     iget-object v1, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/view/ViewGroup;
  
!     invoke-virtual {v1}, Landroid/view/ViewGroup;->getParent()Landroid/view/ViewParent;
  
      move-result-object v1
  
***************
*** 1628,1639 ****
  
      iget-object v1, p0, Lcom/android/internal/widget/ActionBarView;->mUpGoerFive:Landroid/view/ViewGroup;
  
!     iget-object v2, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/widget/LinearLayout;
  
      invoke-virtual {v1, v2}, Landroid/view/ViewGroup;->removeView(Landroid/view/View;)V
  
      :cond_0
!     iput-object v3, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/widget/LinearLayout;
  
      iget v1, p0, Lcom/android/internal/widget/ActionBarView;->mDisplayOptions:I
  
--- 1744,1755 ----
  
      iget-object v1, p0, Lcom/android/internal/widget/ActionBarView;->mUpGoerFive:Landroid/view/ViewGroup;
  
!     iget-object v2, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/view/ViewGroup;
  
      invoke-virtual {v1, v2}, Landroid/view/ViewGroup;->removeView(Landroid/view/View;)V
  
      :cond_0
!     iput-object v3, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/view/ViewGroup;
  
      iget v1, p0, Lcom/android/internal/widget/ActionBarView;->mDisplayOptions:I
  
***************
*** 1856,1870 ****
      :goto_4
      move-object/from16 v0, p0
  
!     iget-object v3, v0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/widget/LinearLayout;
  
      if-eqz v3, :cond_f
  
      move-object/from16 v0, p0
  
!     iget-object v3, v0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/widget/LinearLayout;
  
!     invoke-virtual {v3}, Landroid/widget/LinearLayout;->getVisibility()I
  
      move-result v3
  
--- 1972,1986 ----
      :goto_4
      move-object/from16 v0, p0
  
!     iget-object v3, v0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/view/ViewGroup;
  
      if-eqz v3, :cond_f
  
      move-object/from16 v0, p0
  
!     iget-object v3, v0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/view/ViewGroup;
  
!     invoke-virtual {v3}, Landroid/view/ViewGroup;->getVisibility()I
  
      move-result v3
  
***************
*** 2705,2710 ****
--- 2821,2834 ----
  
      move/from16 v45, v0
  
+     move-object/from16 v0, p0
+ 
+     move/from16 v1, v45
+ 
+     invoke-static {v0, v1}, Lcom/android/internal/widget/Injector$ActionBarViewHook;->isCollapsable(Lcom/android/internal/widget/ActionBarView;Z)Z
+ 
+     move-result v45
+ 
      if-eqz v45, :cond_6
  
      const/16 v43, 0x0
***************
*** 2755,2761 ****
  
      move-object/from16 v45, v0
  
!     invoke-virtual/range {v45 .. v45}, Lcom/android/internal/view/menu/ActionMenuView;->getChildCount()I
  
      move-result v45
  
--- 2879,2885 ----
  
      move-object/from16 v45, v0
  
!     invoke-static/range {v45 .. v45}, Lcom/android/internal/widget/Injector$ActionBarViewHook;->getMenuItemCount(Lcom/android/internal/view/menu/ActionMenuView;)I
  
      move-result v45
  
***************
*** 3059,3065 ****
      .local v36, rightOfCenter:I
      move-object/from16 v0, p0
  
!     iget-object v0, v0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/widget/LinearLayout;
  
      move-object/from16 v45, v0
  
--- 3183,3189 ----
      .local v36, rightOfCenter:I
      move-object/from16 v0, p0
  
!     iget-object v0, v0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/view/ViewGroup;
  
      move-object/from16 v45, v0
  
***************
*** 3067,3077 ****
  
      move-object/from16 v0, p0
  
!     iget-object v0, v0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/widget/LinearLayout;
  
      move-object/from16 v45, v0
  
!     invoke-virtual/range {v45 .. v45}, Landroid/widget/LinearLayout;->getVisibility()I
  
      move-result v45
  
--- 3191,3201 ----
  
      move-object/from16 v0, p0
  
!     iget-object v0, v0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/view/ViewGroup;
  
      move-object/from16 v45, v0
  
!     invoke-virtual/range {v45 .. v45}, Landroid/view/ViewGroup;->getVisibility()I
  
      move-result v45
  
***************
*** 3639,3645 ****
  
      move-object/from16 v0, p0
  
!     iget-object v0, v0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/widget/LinearLayout;
  
      move-object/from16 v45, v0
  
--- 3763,3769 ----
  
      move-object/from16 v0, p0
  
!     iget-object v0, v0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/view/ViewGroup;
  
      move-object/from16 v45, v0
  
***************
*** 3649,3659 ****
  
      move-object/from16 v0, p0
  
!     iget-object v0, v0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/widget/LinearLayout;
  
      move-object/from16 v46, v0
  
!     invoke-virtual/range {v46 .. v46}, Landroid/widget/LinearLayout;->getMeasuredWidth()I
  
      move-result v46
  
--- 3773,3783 ----
  
      move-object/from16 v0, p0
  
!     iget-object v0, v0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/view/ViewGroup;
  
      move-object/from16 v46, v0
  
!     invoke-virtual/range {v46 .. v46}, Landroid/view/ViewGroup;->getMeasuredWidth()I
  
      move-result v46
  
***************
*** 4288,4293 ****
--- 4412,4457 ----
      return-object v0
  .end method
  
+ .method public onWindowHide()V
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/widget/ActionBarView;->mSplitView:Lcom/android/internal/widget/ActionBarContainer;
+ 
+     instance-of v0, v0, Lcom/miui/internal/v5/widget/ActionBarContainer;
+ 
+     if-eqz v0, :cond_0
+ 
+     iget-object v0, p0, Lcom/android/internal/widget/ActionBarView;->mSplitView:Lcom/android/internal/widget/ActionBarContainer;
+ 
+     check-cast v0, Lcom/miui/internal/v5/widget/ActionBarContainer;
+ 
+     invoke-virtual {v0}, Lcom/miui/internal/v5/widget/ActionBarContainer;->onWindowHide()V
+ 
+     :cond_0
+     return-void
+ .end method
+ 
+ .method public onWindowShow()V
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/widget/ActionBarView;->mSplitView:Lcom/android/internal/widget/ActionBarContainer;
+ 
+     instance-of v0, v0, Lcom/miui/internal/v5/widget/ActionBarContainer;
+ 
+     if-eqz v0, :cond_0
+ 
+     iget-object v0, p0, Lcom/android/internal/widget/ActionBarView;->mSplitView:Lcom/android/internal/widget/ActionBarContainer;
+ 
+     check-cast v0, Lcom/miui/internal/v5/widget/ActionBarContainer;
+ 
+     invoke-virtual {v0}, Lcom/miui/internal/v5/widget/ActionBarContainer;->onWindowShow()V
+ 
+     :cond_0
+     return-void
+ .end method
+ 
  .method public setCallback(Landroid/app/ActionBar$OnNavigationListener;)V
      .locals 0
      .parameter "callback"
***************
*** 4522,4528 ****
  
      :cond_4
      :goto_9
!     iget-object v9, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/widget/LinearLayout;
  
      if-eqz v9, :cond_5
  
--- 4686,4698 ----
  
      :cond_4
      :goto_9
!     iget-object v9, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/view/ViewGroup;
! 
!     iget-object v10, p0, Lcom/android/internal/widget/ActionBarView;->mTitleView:Landroid/widget/TextView;
! 
!     invoke-static {v9, v10}, Lcom/android/internal/widget/Injector$ActionBarViewHook;->checkTitleLayout(Landroid/view/View;Landroid/view/View;)Z
! 
!     move-result v9
  
      if-eqz v9, :cond_5
  
***************
*** 4592,4598 ****
      :cond_a
      iget-object v9, p0, Lcom/android/internal/widget/ActionBarView;->mUpGoerFive:Landroid/view/ViewGroup;
  
!     iget-object v10, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/widget/LinearLayout;
  
      invoke-virtual {v9, v10}, Landroid/view/ViewGroup;->removeView(Landroid/view/View;)V
  
--- 4762,4768 ----
      :cond_a
      iget-object v9, p0, Lcom/android/internal/widget/ActionBarView;->mUpGoerFive:Landroid/view/ViewGroup;
  
!     iget-object v10, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/view/ViewGroup;
  
      invoke-virtual {v9, v10}, Landroid/view/ViewGroup;->removeView(Landroid/view/View;)V
  
***************
*** 5032,5042 ****
  
      if-nez v4, :cond_3
  
!     new-instance v4, Lcom/android/internal/view/menu/ActionMenuPresenter;
  
!     iget-object v5, p0, Lcom/android/internal/widget/ActionBarView;->mContext:Landroid/content/Context;
  
!     invoke-direct {v4, v5}, Lcom/android/internal/view/menu/ActionMenuPresenter;-><init>(Landroid/content/Context;)V
  
      iput-object v4, p0, Lcom/android/internal/widget/ActionBarView;->mActionMenuPresenter:Lcom/android/internal/view/menu/ActionMenuPresenter;
  
--- 5202,5212 ----
  
      if-nez v4, :cond_3
  
!     iget-object v4, p0, Lcom/android/internal/widget/ActionBarView;->mContext:Landroid/content/Context;
  
!     invoke-static {v4}, Lcom/android/internal/widget/Injector$ActionBarViewHook;->createActionMenuPresenter(Landroid/content/Context;)Lcom/android/internal/view/menu/ActionMenuPresenter;
  
!     move-result-object v4
  
      iput-object v4, p0, Lcom/android/internal/widget/ActionBarView;->mActionMenuPresenter:Lcom/android/internal/view/menu/ActionMenuPresenter;
  
***************
*** 5599,5610 ****
  
      .local v0, visible:Z
      :goto_1
!     iget-object v1, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/widget/LinearLayout;
  
      if-eqz v0, :cond_4
  
      :goto_2
!     invoke-virtual {v1, v2}, Landroid/widget/LinearLayout;->setVisibility(I)V
  
      .end local v0           #visible:Z
      :cond_1
--- 5769,5780 ----
  
      .local v0, visible:Z
      :goto_1
!     iget-object v1, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/view/ViewGroup;
  
      if-eqz v0, :cond_4
  
      :goto_2
!     invoke-virtual {v1, v2}, Landroid/view/ViewGroup;->setVisibility(I)V
  
      .end local v0           #visible:Z
      :cond_1
***************
*** 5635,5640 ****
--- 5805,5820 ----
      goto :goto_2
  .end method
  
+ .method public setSubtitleView(Landroid/widget/TextView;)V
+     .locals 0
+     .parameter "subTitleView"
+ 
+     .prologue
+     iput-object p1, p0, Lcom/android/internal/widget/ActionBarView;->mSubtitleView:Landroid/widget/TextView;
+ 
+     return-void
+ .end method
+ 
  .method public setTitle(Ljava/lang/CharSequence;)V
      .locals 1
      .parameter "title"
***************
*** 5648,5653 ****
--- 5828,5853 ----
  
      return-void
  .end method
+ 
+ .method public setTitleLayout(Landroid/view/ViewGroup;)V
+     .locals 0
+     .parameter "titleLayout"
+ 
+     .prologue
+     iput-object p1, p0, Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/view/ViewGroup;
+ 
+     return-void
+ .end method
+ 
+ .method public setTitleView(Landroid/widget/TextView;)V
+     .locals 0
+     .parameter "titleView"
+ 
+     .prologue
+     iput-object p1, p0, Lcom/android/internal/widget/ActionBarView;->mTitleView:Landroid/widget/TextView;
+ 
+     return-void
+ .end method
  
  .method public setWindowCallback(Landroid/view/Window$Callback;)V
      .locals 0
