*** ./smali/com/android/internal/widget/ILockSettings$Stub$Proxy.smali	2014-08-20 05:49:27.544015546 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/framework2.jar.out/./smali/com/android/internal/widget/ILockSettings$Stub$Proxy.smali	2014-08-20 05:52:12.732019830 +0700
***************
*** 860,865 ****
--- 860,923 ----
      throw v2
  .end method
  
+ .method public setRawLockPassword([BI)V
+     .locals 5
+     .parameter "hash"
+     .parameter "userId"
+     .annotation system Ldalvik/annotation/Throws;
+         value = {
+             Landroid/os/RemoteException;
+         }
+     .end annotation
+ 
+     .prologue
+     invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
+ 
+     move-result-object v0
+ 
+     .local v0, _data:Landroid/os/Parcel;
+     invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
+ 
+     move-result-object v1
+ 
+     .local v1, _reply:Landroid/os/Parcel;
+     :try_start_0
+     const-string v2, "com.android.internal.widget.ILockSettings"
+ 
+     invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
+ 
+     invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeByteArray([B)V
+ 
+     invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeInt(I)V
+ 
+     iget-object v2, p0, Lcom/android/internal/widget/ILockSettings$Stub$Proxy;->mRemote:Landroid/os/IBinder;
+ 
+     const/16 v3, 0x2a
+ 
+     const/4 v4, 0x0
+ 
+     invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
+ 
+     invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
+     :try_end_0
+     .catchall {:try_start_0 .. :try_end_0} :catchall_0
+ 
+     invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
+ 
+     invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
+ 
+     return-void
+ 
+     :catchall_0
+     move-exception v2
+ 
+     invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
+ 
+     invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
+ 
+     throw v2
+ .end method
+ 
  .method public setString(Ljava/lang/String;Ljava/lang/String;I)V
      .locals 5
      .parameter "key"
***************
*** 919,922 ****
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
! .end method
--- 977,980 ----
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
! .end method
\ No newline at end of file
