*** ./smali/android/content/pm/PackageManager.smali	2014-08-20 05:47:29.480012485 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/framework.jar.out/./smali/android/content/pm/PackageManager.smali	2014-08-20 05:50:14.348016760 +0700
***************
*** 16,21 ****
--- 16,23 ----
  
  .field public static final ACTION_REQUEST_PERMISSION:Ljava/lang/String; = "android.content.pm.action.REQUEST_PERMISSION"
  
+ .field public static final COMPONENT_ENABLED_STATE_ACCESS_CONTROL:I = -0x80000000
+ 
  .field public static final COMPONENT_ENABLED_STATE_DEFAULT:I = 0x0
  
  .field public static final COMPONENT_ENABLED_STATE_DISABLED:I = 0x2
***************
*** 24,31 ****
--- 26,37 ----
  
  .field public static final COMPONENT_ENABLED_STATE_DISABLED_USER:I = 0x3
  
+ .field public static final COMPONENT_ENABLED_STATE_DISABLE_AUTOSTART:I = 0x40000000
+ 
  .field public static final COMPONENT_ENABLED_STATE_ENABLED:I = 0x1
  
+ .field public static final COMPONENT_ENABLED_STATE_SHOW_FLOATING_WINDOW:I = 0x8000000
+ 
  .field public static final DELETE_ALL_USERS:I = 0x2
  
  .field public static final DELETE_FAILED_DEVICE_POLICY_MANAGER:I = -0x2
***************
*** 190,195 ****
--- 196,205 ----
  
  .field public static final GET_URI_PERMISSION_PATTERNS:I = 0x800
  
+ .field public static final HAS_ACTIVITY:I = 0x20000
+ 
+ .field public static final HAS_ACTIVITY_OR_SERVICES:I = 0x40000
+ 
  .field public static final INSTALL_ALLOW_DOWNGRADE:I = 0x80
  
  .field public static final INSTALL_ALLOW_TEST:I = 0x4
***************
*** 204,209 ****
--- 214,221 ----
  
  .field public static final INSTALL_FAILED_CONTAINER_ERROR:I = -0x12
  
+ .field public static final INSTALL_FAILED_CONTAIN_VIRUS:I = -0x19
+ 
  .field public static final INSTALL_FAILED_CPU_ABI_INCOMPATIBLE:I = -0x10
  
  .field public static final INSTALL_FAILED_DEXOPT:I = -0xb
***************
*** 238,243 ****
--- 250,257 ----
  
  .field public static final INSTALL_FAILED_SHARED_USER_INCOMPATIBLE:I = -0x8
  
+ .field public static final INSTALL_FAILED_SYSTEM_INCOMPATIBLE:I = -0x1a
+ 
  .field public static final INSTALL_FAILED_TEST_ONLY:I = -0xf
  
  .field public static final INSTALL_FAILED_UID_CHANGED:I = -0x18
***************
*** 256,261 ****
--- 270,277 ----
  
  .field public static final INSTALL_FROM_ADB:I = 0x20
  
+ .field public static final INSTALL_FROM_XIAOMI:I = 0x100
+ 
  .field public static final INSTALL_INTERNAL:I = 0x10
  
  .field public static final INSTALL_PARSE_FAILED_BAD_MANIFEST:I = -0x65
***************
*** 935,940 ****
--- 951,974 ----
      .end annotation
  .end method
  
+ .method public getUsers()Ljava/util/List;
+     .locals 1
+     .annotation system Ldalvik/annotation/Signature;
+         value = {
+             "()",
+             "Ljava/util/List",
+             "<",
+             "Landroid/content/pm/UserInfo;",
+             ">;"
+         }
+     .end annotation
+ 
+     .prologue
+     const/4 v0, 0x0
+ 
+     return-object v0
+ .end method
+ 
  .method public abstract getVerifierDeviceIdentity()Landroid/content/pm/VerifierDeviceIdentity;
  .end method
  
