*** ./smali/com/android/internal/os/ZygoteInit.smali	2014-08-20 05:49:29.960015609 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/framework2.jar.out/./smali/com/android/internal/os/ZygoteInit.smali	2014-08-20 05:52:15.140019892 +0700
***************
*** 425,430 ****
--- 425,432 ----
      :try_start_0
      invoke-static {}, Lcom/android/internal/os/SamplingProfilerIntegration;->start()V
  
+     invoke-static {}, Lmiui/security/SecurityManager;->init()V
+ 
      invoke-static {}, Lcom/android/internal/os/ZygoteInit;->registerZygoteSocket()V
  
      const/16 v2, 0xbcc
***************
*** 1275,1280 ****
--- 1277,1358 ----
      return v0
  .end method
  
+ .method private static preloadMiuiResources()V
+     .locals 8
+ 
+     .prologue
+     invoke-static {}, Landroid/os/SystemClock;->uptimeMillis()J
+ 
+     move-result-wide v2
+ 
+     .local v2, startTime:J
+     sget-object v4, Lcom/android/internal/os/ZygoteInit;->mResources:Landroid/content/res/Resources;
+ 
+     const v5, 0x6060013
+ 
+     invoke-virtual {v4, v5}, Landroid/content/res/Resources;->obtainTypedArray(I)Landroid/content/res/TypedArray;
+ 
+     move-result-object v1
+ 
+     .local v1, ar:Landroid/content/res/TypedArray;
+     invoke-static {}, Ldalvik/system/VMRuntime;->getRuntime()Ldalvik/system/VMRuntime;
+ 
+     move-result-object v4
+ 
+     invoke-static {v4, v1}, Lcom/android/internal/os/ZygoteInit;->preloadDrawables(Ldalvik/system/VMRuntime;Landroid/content/res/TypedArray;)I
+ 
+     move-result v0
+ 
+     .local v0, N:I
+     invoke-virtual {v1}, Landroid/content/res/TypedArray;->recycle()V
+ 
+     const-string v4, "Zygote"
+ 
+     new-instance v5, Ljava/lang/StringBuilder;
+ 
+     invoke-direct {v5}, Ljava/lang/StringBuilder;-><init>()V
+ 
+     const-string v6, "...preloaded "
+ 
+     invoke-virtual {v5, v6}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v5
+ 
+     invoke-virtual {v5, v0}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
+ 
+     move-result-object v5
+ 
+     const-string v6, " miui resources in "
+ 
+     invoke-virtual {v5, v6}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v5
+ 
+     invoke-static {}, Landroid/os/SystemClock;->uptimeMillis()J
+ 
+     move-result-wide v6
+ 
+     sub-long/2addr v6, v2
+ 
+     invoke-virtual {v5, v6, v7}, Ljava/lang/StringBuilder;->append(J)Ljava/lang/StringBuilder;
+ 
+     move-result-object v5
+ 
+     const-string v6, "ms."
+ 
+     invoke-virtual {v5, v6}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v5
+ 
+     invoke-virtual {v5}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
+ 
+     move-result-object v5
+ 
+     invoke-static {v4, v5}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
+ 
+     return-void
+ .end method
+ 
  .method private static preloadOpenGL()V
      .locals 2
  
***************
*** 1392,1397 ****
--- 1470,1477 ----
  
      invoke-static {v6, v7}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
  
+     invoke-static {}, Lcom/android/internal/os/ZygoteInit;->preloadMiuiResources()V
+ 
      invoke-static {}, Landroid/os/SystemClock;->uptimeMillis()J
  
      move-result-wide v4
