*** ./smali/com/android/internal/telephony/uicc/IccUtils.smali	2014-08-20 05:50:07.444016581 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/telephony-common.jar.out/./smali/com/android/internal/telephony/uicc/IccUtils.smali	2014-08-20 05:52:52.752020867 +0700
***************
*** 1171,1176 ****
--- 1171,1268 ----
      .end packed-switch
  .end method
  
+ .method public static parseIccIdToString([BII)Ljava/lang/String;
+     .locals 5
+     .parameter "data"
+     .parameter "offset"
+     .parameter "length"
+ 
+     .prologue
+     const/16 v4, 0x9
+ 
+     new-instance v1, Ljava/lang/StringBuilder;
+ 
+     mul-int/lit8 v3, p2, 0x2
+ 
+     invoke-direct {v1, v3}, Ljava/lang/StringBuilder;-><init>(I)V
+ 
+     .local v1, ret:Ljava/lang/StringBuilder;
+     move v0, p1
+ 
+     .local v0, i:I
+     :goto_0
+     add-int v3, p1, p2
+ 
+     if-ge v0, v3, :cond_2
+ 
+     aget-byte v3, p0, v0
+ 
+     and-int/lit8 v2, v3, 0xf
+ 
+     .local v2, v:I
+     if-ltz v2, :cond_0
+ 
+     if-gt v2, v4, :cond_0
+ 
+     add-int/lit8 v3, v2, 0x30
+ 
+     int-to-char v3, v3
+ 
+     invoke-virtual {v1, v3}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
+ 
+     :goto_1
+     aget-byte v3, p0, v0
+ 
+     shr-int/lit8 v3, v3, 0x4
+ 
+     and-int/lit8 v2, v3, 0xf
+ 
+     if-ltz v2, :cond_1
+ 
+     if-gt v2, v4, :cond_1
+ 
+     add-int/lit8 v3, v2, 0x30
+ 
+     int-to-char v3, v3
+ 
+     invoke-virtual {v1, v3}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
+ 
+     :goto_2
+     add-int/lit8 v0, v0, 0x1
+ 
+     goto :goto_0
+ 
+     :cond_0
+     add-int/lit8 v3, v2, 0x61
+ 
+     add-int/lit8 v3, v3, -0xa
+ 
+     int-to-char v3, v3
+ 
+     invoke-virtual {v1, v3}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
+ 
+     goto :goto_1
+ 
+     :cond_1
+     add-int/lit8 v3, v2, 0x61
+ 
+     add-int/lit8 v3, v3, -0xa
+ 
+     int-to-char v3, v3
+ 
+     invoke-virtual {v1, v3}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
+ 
+     goto :goto_2
+ 
+     .end local v2           #v:I
+     :cond_2
+     invoke-virtual {v1}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
+ 
+     move-result-object v3
+ 
+     return-object v3
+ .end method
+ 
  .method public static parseToBnW([BI)Landroid/graphics/Bitmap;
      .locals 13
      .parameter "data"
