*** ./smali/android/util/TypedValue.smali	2014-08-20 05:49:11.468015129 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/framework.jar.out/./smali/android/util/TypedValue.smali	2014-08-20 05:51:56.636019412 +0700
***************
*** 88,93 ****
--- 88,95 ----
  
  .field public static final TYPE_STRING:I = 0x3
  
+ .field public static miui_font_scale:F
+ 
  
  # instance fields
  .field public assetCookie:I
***************
*** 118,123 ****
--- 120,129 ----
  
      const/4 v3, 0x0
  
+     const/high16 v0, 0x3f80
+ 
+     sput v0, Landroid/util/TypedValue;->miui_font_scale:F
+ 
      new-array v0, v6, [F
  
      fill-array-data v0, :array_0
***************
*** 219,227 ****
      goto :goto_0
  
      :pswitch_2
!     iget v0, p2, Landroid/util/DisplayMetrics;->scaledDensity:F
  
!     mul-float/2addr p1, v0
  
      goto :goto_0
  
--- 225,233 ----
      goto :goto_0
  
      :pswitch_2
!     invoke-static {p0, p1, p2}, Landroid/util/Injector$TypedValueHook;->miuiScale(IFLandroid/util/DisplayMetrics;)F
  
!     move-result p1
  
      goto :goto_0
  
