*** ./smali/android/preference/PreferenceScreen.smali	2014-08-20 05:48:01.700013320 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/framework.jar.out/./smali/android/preference/PreferenceScreen.smali	2014-08-20 05:50:46.552017595 +0700
***************
*** 93,105 ****
      move-result-object v4
  
      .local v4, title:Ljava/lang/CharSequence;
!     new-instance v2, Landroid/app/Dialog;
  
      invoke-virtual {v1}, Landroid/content/Context;->getThemeResId()I
  
      move-result v5
  
!     invoke-direct {v2, v1, v5}, Landroid/app/Dialog;-><init>(Landroid/content/Context;I)V
  
      iput-object v2, p0, Landroid/preference/PreferenceScreen;->mDialog:Landroid/app/Dialog;
  
--- 93,105 ----
      move-result-object v4
  
      .local v4, title:Ljava/lang/CharSequence;
!     new-instance v2, Landroid/app/PreferenceDialog;
  
      invoke-virtual {v1}, Landroid/content/Context;->getThemeResId()I
  
      move-result v5
  
!     invoke-direct {v2, v1, v5}, Landroid/app/PreferenceDialog;-><init>(Landroid/content/Context;I)V
  
      iput-object v2, p0, Landroid/preference/PreferenceScreen;->mDialog:Landroid/app/Dialog;
  
***************
*** 157,164 ****
--- 157,172 ----
  
      move-result-object v0
  
+     invoke-static {p1, v0}, Landroid/preference/Injector$PreferenceScreenHook;->addHeaderView(Landroid/widget/ListView;Landroid/widget/ListAdapter;)V
+ 
+     invoke-virtual {p0}, Landroid/preference/PreferenceScreen;->getRootAdapter()Landroid/widget/ListAdapter;
+ 
+     move-result-object v0
+ 
      invoke-virtual {p1, v0}, Landroid/widget/ListView;->setAdapter(Landroid/widget/ListAdapter;)V
  
+     invoke-static {p1}, Landroid/preference/Injector$PreferenceScreenHook;->bind(Landroid/widget/ListView;)V
+ 
      invoke-virtual {p0}, Landroid/preference/PreferenceScreen;->onAttachedToActivity()V
  
      return-void
