*** ./smali/android/preference/RingtonePreference.smali	2014-08-20 05:48:01.264013309 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/framework.jar.out/./smali/android/preference/RingtonePreference.smali	2014-08-20 05:50:46.124017584 +0700
***************
*** 311,316 ****
--- 311,318 ----
  
      invoke-virtual {p1, v0, v1}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Ljava/lang/CharSequence;)Landroid/content/Intent;
  
+     invoke-static {p1}, Landroid/preference/Injector$RingtonePreferenceHook;->after_onPrepareRingtonePickerIntent(Landroid/content/Intent;)V
+ 
      return-void
  .end method
  
