*** ./smali/com/android/internal/telephony/dataconnection/DcTracker.smali	2014-08-20 05:50:03.896016489 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/telephony-common.jar.out/./smali/com/android/internal/telephony/dataconnection/DcTracker.smali	2014-08-20 05:52:49.196020775 +0700
***************
*** 8448,8453 ****
--- 8448,8461 ----
      .end local v16           #proxy:Landroid/net/ProxyProperties;
      :cond_4
      :goto_2
+     invoke-virtual {v3}, Lcom/android/internal/telephony/dataconnection/ApnContext;->getApnSetting()Lcom/android/internal/telephony/dataconnection/ApnSetting;
+ 
+     move-result-object v17
+ 
+     move-object/from16 v0, v17
+ 
+     invoke-static {v3, v0}, Lcom/android/internal/telephony/dataconnection/Injector$DcTrackerHook;->onDataSetupComplete(Lcom/android/internal/telephony/dataconnection/ApnContext;Lcom/android/internal/telephony/dataconnection/ApnSetting;)V
+ 
      invoke-virtual {v3}, Lcom/android/internal/telephony/dataconnection/ApnContext;->getApnType()Ljava/lang/String;
  
      move-result-object v17
***************
*** 9256,9261 ****
--- 9264,9275 ----
  
      invoke-virtual {v0, v2}, Lcom/android/internal/telephony/dataconnection/ApnContext;->setState(Lcom/android/internal/telephony/DctConstants$State;)V
  
+     invoke-virtual {v0}, Lcom/android/internal/telephony/dataconnection/ApnContext;->getApnSetting()Lcom/android/internal/telephony/dataconnection/ApnSetting;
+ 
+     move-result-object v2
+ 
+     invoke-static {v0, v2}, Lcom/android/internal/telephony/dataconnection/Injector$DcTrackerHook;->onDisconnectDone(Lcom/android/internal/telephony/dataconnection/ApnContext;Lcom/android/internal/telephony/dataconnection/ApnSetting;)V
+ 
      iget-object v2, p0, Lcom/android/internal/telephony/dataconnection/DcTracker;->mPhone:Lcom/android/internal/telephony/PhoneBase;
  
      invoke-virtual {v0}, Lcom/android/internal/telephony/dataconnection/ApnContext;->getReason()Ljava/lang/String;
