*** ./smali/android/content/ContentResolver.smali	2014-08-20 05:47:33.264012583 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/framework.jar.out/./smali/android/content/ContentResolver.smali	2014-08-20 05:50:17.944016853 +0700
***************
*** 4084,4090 ****
  .end method
  
  .method public final query(Landroid/net/Uri;[Ljava/lang/String;Ljava/lang/String;[Ljava/lang/String;Ljava/lang/String;Landroid/os/CancellationSignal;)Landroid/database/Cursor;
!     .locals 24
      .parameter "uri"
      .parameter "projection"
      .parameter "selection"
--- 4084,4090 ----
  .end method
  
  .method public final query(Landroid/net/Uri;[Ljava/lang/String;Ljava/lang/String;[Ljava/lang/String;Ljava/lang/String;Landroid/os/CancellationSignal;)Landroid/database/Cursor;
!     .locals 28
      .parameter "uri"
      .parameter "projection"
      .parameter "selection"
***************
*** 4135,4140 ****
--- 4135,4141 ----
      invoke-virtual {v0, v10}, Landroid/os/CancellationSignal;->setRemote(Landroid/os/ICancellationSignal;)V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
+     .catch Ljava/lang/IllegalStateException; {:try_start_0 .. :try_end_0} :catch_miui
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_1
  
      :cond_2
***************
*** 4157,4162 ****
--- 4158,4164 ----
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
      .catch Landroid/os/DeadObjectException; {:try_start_1 .. :try_end_1} :catch_0
+     .catch Ljava/lang/IllegalStateException; {:try_start_1 .. :try_end_1} :catch_miui
      .catch Landroid/os/RemoteException; {:try_start_1 .. :try_end_1} :catch_1
  
      move-result-object v20
***************
*** 4189,4194 ****
--- 4191,4197 ----
      :cond_5
      if-eqz v11, :cond_0
  
+     :goto_miui_1
      move-object/from16 v0, p0
  
      invoke-virtual {v0, v11}, Landroid/content/ContentResolver;->releaseProvider(Landroid/content/IContentProvider;)Z
***************
*** 4207,4212 ****
--- 4210,4216 ----
      invoke-virtual/range {p0 .. p1}, Landroid/content/ContentResolver;->acquireProvider(Landroid/net/Uri;)Landroid/content/IContentProvider;
      :try_end_2
      .catchall {:try_start_2 .. :try_end_2} :catchall_0
+     .catch Ljava/lang/IllegalStateException; {:try_start_2 .. :try_end_2} :catch_miui
      .catch Landroid/os/RemoteException; {:try_start_2 .. :try_end_2} :catch_1
  
      move-result-object v11
***************
*** 4307,4312 ****
--- 4311,4317 ----
      invoke-direct {v0, v1, v2, v4}, Landroid/content/ContentResolver$CursorWrapperInner;-><init>(Landroid/content/ContentResolver;Landroid/database/Cursor;Landroid/content/IContentProvider;)V
      :try_end_3
      .catchall {:try_start_3 .. :try_end_3} :catchall_0
+     .catch Ljava/lang/IllegalStateException; {:try_start_3 .. :try_end_3} :catch_miui
      .catch Landroid/os/RemoteException; {:try_start_3 .. :try_end_3} :catch_1
  
      .local v23, wrapper:Landroid/content/ContentResolver$CursorWrapperInner;
***************
*** 4349,4354 ****
--- 4354,4360 ----
      invoke-virtual/range {p0 .. p1}, Landroid/content/ContentResolver;->acquireProvider(Landroid/net/Uri;)Landroid/content/IContentProvider;
      :try_end_4
      .catchall {:try_start_4 .. :try_end_4} :catchall_0
+     .catch Ljava/lang/IllegalStateException; {:try_start_4 .. :try_end_4} :catch_miui
      .catch Landroid/os/RemoteException; {:try_start_4 .. :try_end_4} :catch_1
  
      move-result-object v4
***************
*** 4394,4399 ****
--- 4400,4460 ----
      goto/16 :goto_0
  
      .end local v19           #e:Landroid/os/RemoteException;
+     :catch_miui
+     move-exception v16
+ 
+     .local v16, e:Ljava/lang/IllegalStateException;
+     :try_start_miui
+     move-object/from16 v0, p0
+ 
+     iget-object v0, v0, Landroid/content/ContentResolver;->mContext:Landroid/content/Context;
+ 
+     move-object/from16 v21, v0
+ 
+     move-object/from16 v22, v16
+ 
+     move-object/from16 v23, p1
+ 
+     move-object/from16 v24, p2
+ 
+     move-object/from16 v25, p3
+ 
+     move-object/from16 v26, p4
+ 
+     move-object/from16 v27, p5
+ 
+     invoke-static/range {v21 .. v27}, Landroid/content/Injector$ContentResolverHook;->checkCursorLeakException(Landroid/content/Context;Ljava/lang/IllegalStateException;Landroid/net/Uri;[Ljava/lang/String;Ljava/lang/String;[Ljava/lang/String;Ljava/lang/String;)Landroid/database/Cursor;
+     :try_end_miui
+     .catchall {:try_start_miui .. :try_end_miui} :catchall_0
+ 
+     move-result-object v23
+ 
+     if-eqz v20, :cond_miui_1
+ 
+     invoke-interface/range {v22 .. v22}, Landroid/database/Cursor;->close()V
+ 
+     :cond_miui_1
+     if-eqz p6, :cond_miui_0
+ 
+     const/4 v4, 0x0
+ 
+     move-object/from16 v0, p6
+ 
+     invoke-virtual {v0, v4}, Landroid/os/CancellationSignal;->setRemote(Landroid/os/ICancellationSignal;)V
+ 
+     :cond_miui_0
+     if-eqz v3, :cond_miui_2
+ 
+     move-object/from16 v0, p0
+ 
+     invoke-virtual {v0, v3}, Landroid/content/ContentResolver;->releaseUnstableProvider(Landroid/content/IContentProvider;)Z
+ 
+     :cond_miui_2
+     if-eqz v11, :cond_0
+ 
+     goto/16 :goto_miui_1
+ 
+     .end local v16           #e:Ljava/lang/IllegalStateException;
      :catchall_0
      move-exception v4
  
