*** ./smali/com/android/internal/view/menu/ActionMenuItemView.smali	2014-08-20 05:49:21.460015388 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/framework2.jar.out/./smali/com/android/internal/view/menu/ActionMenuItemView.smali	2014-08-20 05:52:06.680019673 +0700
***************
*** 32,37 ****
--- 32,39 ----
  
  .field private mSavedPaddingLeft:I
  
+ .field mSavedPaddingRight:I
+ 
  .field private mTitle:Ljava/lang/CharSequence;
  
  
***************
*** 156,161 ****
--- 158,171 ----
  
      iget-object v3, p0, Lcom/android/internal/view/menu/ActionMenuItemView;->mItemData:Lcom/android/internal/view/menu/MenuItemImpl;
  
+     invoke-virtual {v3}, Lcom/android/internal/view/menu/MenuItemImpl;->isForceShowText()Z
+ 
+     move-result v3
+ 
+     if-nez v3, :cond_0
+ 
+     iget-object v3, p0, Lcom/android/internal/view/menu/ActionMenuItemView;->mItemData:Lcom/android/internal/view/menu/MenuItemImpl;
+ 
      invoke-virtual {v3}, Lcom/android/internal/view/menu/MenuItemImpl;->showsTextAsAction()Z
  
      move-result v3
***************
*** 224,229 ****
--- 234,248 ----
      return v0
  .end method
  
+ .method getIcon()Landroid/graphics/drawable/Drawable;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/view/menu/ActionMenuItemView;->mIcon:Landroid/graphics/drawable/Drawable;
+ 
+     return-object v0
+ .end method
+ 
  .method public getItemData()Lcom/android/internal/view/menu/MenuItemImpl;
      .locals 1
  
***************
*** 233,238 ****
--- 252,266 ----
      return-object v0
  .end method
  
+ .method getSavedPaddingLeft()I
+     .locals 1
+ 
+     .prologue
+     iget v0, p0, Lcom/android/internal/view/menu/ActionMenuItemView;->mSavedPaddingLeft:I
+ 
+     return v0
+ .end method
+ 
  .method public hasText()Z
      .locals 1
  
***************
*** 516,521 ****
--- 544,558 ----
  
      const/high16 v11, 0x4000
  
+     invoke-static {p0, p1, p2}, Lcom/android/internal/view/menu/Injector$ActionMenuItemViewHook;->before_onMeasure(Lcom/android/internal/view/menu/ActionMenuItemView;II)Z
+ 
+     move-result v7
+ 
+     if-eqz v7, :cond_miui
+ 
+     return-void
+ 
+     :cond_miui
      invoke-static {p2}, Landroid/view/View$MeasureSpec;->getMode(I)I
  
      move-result v7
***************
*** 830,835 ****
--- 867,874 ----
      .prologue
      iput p1, p0, Lcom/android/internal/view/menu/ActionMenuItemView;->mSavedPaddingLeft:I
  
+     iput p3, p0, Lcom/android/internal/view/menu/ActionMenuItemView;->mSavedPaddingRight:I
+ 
      invoke-super {p0, p1, p2, p3, p4}, Landroid/widget/TextView;->setPadding(IIII)V
  
      return-void
***************
*** 868,870 ****
--- 907,933 ----
  
      return v0
  .end method
+ 
+ .method superOnMeasure(II)V
+     .locals 0
+     .parameter "widthMeasureSpec"
+     .parameter "heightMeasureSpec"
+ 
+     .prologue
+     invoke-super {p0, p1, p2}, Landroid/widget/TextView;->onMeasure(II)V
+ 
+     return-void
+ .end method
+ 
+ .method superSetPadding(IIII)V
+     .locals 0
+     .parameter "l"
+     .parameter "t"
+     .parameter "r"
+     .parameter "b"
+ 
+     .prologue
+     invoke-super {p0, p1, p2, p3, p4}, Landroid/widget/TextView;->setPadding(IIII)V
+ 
+     return-void
+ .end method
