*** ./smali/com/android/internal/telephony/gsm/GsmServiceStateTracker.smali	2014-08-20 05:50:07.028016570 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/telephony-common.jar.out/./smali/com/android/internal/telephony/gsm/GsmServiceStateTracker.smali	2014-08-20 05:52:52.332020856 +0700
***************
*** 2382,2388 ****
  
      invoke-virtual {v0, v1, v2}, Lcom/android/internal/telephony/gsm/GSMPhone;->setSystemProperty(Ljava/lang/String;Ljava/lang/String;)V
  
!     if-nez v27, :cond_1e
  
      const-string v37, "operatorNumeric is null"
  
--- 2382,2392 ----
  
      invoke-virtual {v0, v1, v2}, Lcom/android/internal/telephony/gsm/GSMPhone;->setSystemProperty(Ljava/lang/String;Ljava/lang/String;)V
  
!     invoke-static/range {v27 .. v27}, Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z
! 
!     move-result v37
! 
!     if-eqz v37, :cond_1e
  
      const-string v37, "operatorNumeric is null"
  
***************
*** 2960,2965 ****
--- 2964,2979 ----
  
      invoke-direct {v0, v1}, Lcom/android/internal/telephony/gsm/GsmServiceStateTracker;->setAndBroadcastNetworkSetTimeZone(Ljava/lang/String;)V
  
+     invoke-virtual/range {v35 .. v35}, Ljava/util/TimeZone;->getID()Ljava/lang/String;
+ 
+     move-result-object v37
+ 
+     move-object/from16 v0, p0
+ 
+     move-object/from16 v1, v37
+ 
+     invoke-direct {v0, v1}, Lcom/android/internal/telephony/gsm/GsmServiceStateTracker;->saveNitzTimeZone(Ljava/lang/String;)V
+ 
      .end local v30           #testOneUniqueOffsetPath:Z
      .end local v34           #uniqueZones:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Ljava/util/TimeZone;>;"
      :cond_20
***************
*** 8426,8431 ****
--- 8440,8449 ----
  
      .local v6, spn:Ljava/lang/String;
      :goto_3
+     invoke-static {p0, v6}, Lcom/android/internal/telephony/Injector$ServiceStateTrackerHook;->getSpn(Lcom/android/internal/telephony/ServiceStateTracker;Ljava/lang/String;)Ljava/lang/String;
+ 
+     move-result-object v6
+ 
      invoke-static {v6}, Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z
  
      move-result v9
***************
*** 8440,8445 ****
--- 8458,8480 ----
  
      .local v5, showSpn:Z
      :goto_4
+     if-eqz v5, :cond_miui
+ 
+     iget-boolean v9, p0, Lcom/android/internal/telephony/gsm/GsmServiceStateTracker;->mEmergencyOnly:Z
+ 
+     if-nez v9, :cond_miui
+ 
+     iget-object v9, p0, Lcom/android/internal/telephony/gsm/GsmServiceStateTracker;->mSS:Landroid/telephony/ServiceState;
+ 
+     invoke-virtual {v9}, Landroid/telephony/ServiceState;->getState()I
+ 
+     move-result v9
+ 
+     if-nez v9, :cond_miui
+ 
+     move v5, v8
+ 
+     :goto_miui
      iget-boolean v9, p0, Lcom/android/internal/telephony/gsm/GsmServiceStateTracker;->mCurShowPlmn:Z
  
      if-ne v4, v9, :cond_1
***************
*** 8607,8612 ****
--- 8642,8651 ----
  
      move-result-object v2
  
+     invoke-static {p0, v2}, Lcom/android/internal/telephony/Injector$ServiceStateTrackerHook;->getSpn(Lcom/android/internal/telephony/ServiceStateTracker;Ljava/lang/String;)Ljava/lang/String;
+ 
+     move-result-object v2
+ 
      invoke-static {v2}, Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z
  
      move-result v9
***************
*** 8670,8673 ****
--- 8709,8718 ----
      move v5, v7
  
      goto/16 :goto_4
+ 
+     .restart local v5       #showSpn:Z
+     :cond_miui
+     move v5, v7
+ 
+     goto/16 :goto_miui
  .end method
