*** ./smali/com/android/internal/telephony/gsm/SmsMessage.smali	2014-08-20 05:50:06.472016556 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/telephony-common.jar.out/./smali/com/android/internal/telephony/gsm/SmsMessage.smali	2014-08-20 05:52:51.776020842 +0700
***************
*** 2124,2129 ****
--- 2124,2152 ----
      return v0
  .end method
  
+ .method public getRecipientAddress()Ljava/lang/String;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/telephony/gsm/SmsMessage;->mRecipientAddress:Lcom/android/internal/telephony/gsm/GsmSmsAddress;
+ 
+     if-eqz v0, :cond_0
+ 
+     iget-object v0, p0, Lcom/android/internal/telephony/gsm/SmsMessage;->mRecipientAddress:Lcom/android/internal/telephony/gsm/GsmSmsAddress;
+ 
+     invoke-virtual {v0}, Lcom/android/internal/telephony/gsm/GsmSmsAddress;->getAddressString()Ljava/lang/String;
+ 
+     move-result-object v0
+ 
+     :goto_0
+     return-object v0
+ 
+     :cond_0
+     const/4 v0, 0x0
+ 
+     goto :goto_0
+ .end method
+ 
  .method public getStatus()I
      .locals 1
  
