*** ./smali/android/net/http/AndroidHttpClient.smali	2014-08-20 05:48:44.460014429 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/framework.jar.out/./smali/android/net/http/AndroidHttpClient.smali	2014-08-20 05:51:29.600018711 +0700
***************
*** 475,480 ****
--- 475,484 ----
  
      const v8, 0xea60
  
+     invoke-static {p0, p1}, Landroid/net/http/Injector$AndroidHttpClientHook;->before_newInstance(Ljava/lang/String;Landroid/content/Context;)Ljava/lang/String;
+ 
+     move-result-object p0
+ 
      new-instance v1, Lorg/apache/http/params/BasicHttpParams;
  
      invoke-direct {v1}, Lorg/apache/http/params/BasicHttpParams;-><init>()V
