*** ./smali/android/view/Display.smali	2014-08-20 05:47:58.932013248 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/framework.jar.out/./smali/android/view/Display.smali	2014-08-20 05:50:43.824017524 +0700
***************
*** 628,633 ****
--- 628,691 ----
      return v0
  .end method
  
+ .method public getRawHeight()I
+     .locals 1
+ 
+     .prologue
+     monitor-enter p0
+ 
+     :try_start_0
+     invoke-direct {p0}, Landroid/view/Display;->updateDisplayInfoLocked()V
+ 
+     iget-object v0, p0, Landroid/view/Display;->mDisplayInfo:Landroid/view/DisplayInfo;
+ 
+     invoke-virtual {v0}, Landroid/view/DisplayInfo;->getNaturalHeight()I
+ 
+     move-result v0
+ 
+     monitor-exit p0
+ 
+     return v0
+ 
+     :catchall_0
+     move-exception v0
+ 
+     monitor-exit p0
+     :try_end_0
+     .catchall {:try_start_0 .. :try_end_0} :catchall_0
+ 
+     throw v0
+ .end method
+ 
+ .method public getRawWidth()I
+     .locals 1
+ 
+     .prologue
+     monitor-enter p0
+ 
+     :try_start_0
+     invoke-direct {p0}, Landroid/view/Display;->updateDisplayInfoLocked()V
+ 
+     iget-object v0, p0, Landroid/view/Display;->mDisplayInfo:Landroid/view/DisplayInfo;
+ 
+     invoke-virtual {v0}, Landroid/view/DisplayInfo;->getNaturalWidth()I
+ 
+     move-result v0
+ 
+     monitor-exit p0
+ 
+     return v0
+ 
+     :catchall_0
+     move-exception v0
+ 
+     monitor-exit p0
+     :try_end_0
+     .catchall {:try_start_0 .. :try_end_0} :catchall_0
+ 
+     throw v0
+ .end method
+ 
  .method public getRealMetrics(Landroid/util/DisplayMetrics;)V
      .locals 3
      .parameter "outMetrics"
