*** ./smali/android/view/ScaleGestureDetector.smali	2014-08-20 05:47:54.720013139 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/framework.jar.out/./smali/android/view/ScaleGestureDetector.smali	2014-08-20 05:50:39.636017416 +0700
***************
*** 178,183 ****
--- 178,191 ----
      invoke-virtual {p0, v1}, Landroid/view/ScaleGestureDetector;->setQuickScaleEnabled(Z)V
  
      :cond_0
+     const/high16 v2, 0x7fc0
+ 
+     iput v2, p0, Landroid/view/ScaleGestureDetector;->mTouchUpper:F
+ 
+     iput v2, p0, Landroid/view/ScaleGestureDetector;->mTouchLower:F
+ 
+     iput v2, p0, Landroid/view/ScaleGestureDetector;->mTouchHistoryLastAccepted:F
+ 
      return-void
  
      .end local v0           #res:Landroid/content/res/Resources;
