*** ./smali/android/view/ViewConfiguration.smali	2014-08-20 05:47:55.348013155 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/framework.jar.out/./smali/android/view/ViewConfiguration.smali	2014-08-20 05:50:40.264017432 +0700
***************
*** 340,346 ****
  
      iput v10, p0, Landroid/view/ViewConfiguration;->mMaximumDrawingCacheSize:I
  
!     const/4 v10, 0x0
  
      mul-float/2addr v10, v7
  
--- 340,346 ----
  
      iput v10, p0, Landroid/view/ViewConfiguration;->mMaximumDrawingCacheSize:I
  
!     const/high16 v10, 0x40c0
  
      mul-float/2addr v10, v7
  
***************
*** 350,355 ****
--- 350,359 ----
  
      float-to-int v10, v10
  
+     invoke-static {p1, v10}, Landroid/view/Injector$ViewConfigurationHook;->getOverscrollDistance(Landroid/content/Context;I)I
+ 
+     move-result v10
+ 
      iput v10, p0, Landroid/view/ViewConfiguration;->mOverscrollDistance:I
  
      const/high16 v10, 0x40c0
***************
*** 362,367 ****
--- 366,375 ----
  
      float-to-int v10, v10
  
+     invoke-static {p1, v10}, Landroid/view/Injector$ViewConfigurationHook;->getOverflingDistance(Landroid/content/Context;I)I
+ 
+     move-result v10
+ 
      iput v10, p0, Landroid/view/ViewConfiguration;->mOverflingDistance:I
  
      iget-boolean v10, p0, Landroid/view/ViewConfiguration;->sHasPermanentMenuKeySet:Z
***************
*** 452,462 ****
--- 460,491 ----
      goto :goto_2
  .end method
  
+ .method static callConstructor(Landroid/content/Context;)Landroid/view/ViewConfiguration;
+     .locals 1
+     .parameter "context"
+ 
+     .prologue
+     new-instance v0, Landroid/view/ViewConfiguration;
+ 
+     invoke-direct {v0, p0}, Landroid/view/ViewConfiguration;-><init>(Landroid/content/Context;)V
+ 
+     return-object v0
+ .end method
+ 
  .method public static get(Landroid/content/Context;)Landroid/view/ViewConfiguration;
      .locals 5
      .parameter "context"
  
      .prologue
+     invoke-static {p0}, Landroid/view/Injector$ViewConfigurationHook;->get(Landroid/content/Context;)Landroid/view/ViewConfiguration;
+ 
+     move-result-object v0
+ 
+     if-eqz v0, :cond_miui
+ 
+     return-object v0
+ 
+     :cond_miui
      invoke-virtual {p0}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
  
      move-result-object v3
