*** ./smali/android/view/WindowManager$LayoutParams.smali	2014-08-20 05:47:56.908013196 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/framework.jar.out/./smali/android/view/WindowManager$LayoutParams.smali	2014-08-20 05:50:41.804017472 +0700
***************
*** 321,326 ****
--- 321,328 ----
  
  .field public dimAmount:F
  
+ .field public extraFlags:I
+ 
  .field public flags:I
      .annotation runtime Landroid/view/ViewDebug$ExportedProperty;
          flagMapping = {
***************
*** 1106,1111 ****
--- 1108,1119 ----
  
      move-result v0
  
+     iput v0, p0, Landroid/view/WindowManager$LayoutParams;->extraFlags:I
+ 
+     invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
+ 
+     move-result v0
+ 
      iput v0, p0, Landroid/view/WindowManager$LayoutParams;->privateFlags:I
  
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
***************
*** 1487,1492 ****
--- 1495,1513 ----
  
      .end local v1           #diff:I
      :cond_a
+     iget v2, p0, Landroid/view/WindowManager$LayoutParams;->extraFlags:I
+ 
+     iget v3, p1, Landroid/view/WindowManager$LayoutParams;->extraFlags:I
+ 
+     if-eq v2, v3, :cond_miui
+ 
+     iget v2, p1, Landroid/view/WindowManager$LayoutParams;->extraFlags:I
+ 
+     iput v2, p0, Landroid/view/WindowManager$LayoutParams;->extraFlags:I
+ 
+     or-int/lit8 v0, v0, 0x4
+ 
+     :cond_miui
      iget v2, p0, Landroid/view/WindowManager$LayoutParams;->privateFlags:I
  
      iget v3, p1, Landroid/view/WindowManager$LayoutParams;->privateFlags:I
***************
*** 2132,2137 ****
--- 2153,2170 ----
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     const-string v1, " extfl=#"
+ 
+     invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     iget v1, p0, Landroid/view/WindowManager$LayoutParams;->extraFlags:I
+ 
+     invoke-static {v1}, Ljava/lang/Integer;->toHexString(I)Ljava/lang/String;
+ 
+     move-result-object v1
+ 
+     invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
      iget v1, p0, Landroid/view/WindowManager$LayoutParams;->privateFlags:I
  
      if-eqz v1, :cond_5
***************
*** 2427,2432 ****
--- 2460,2469 ----
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     iget v0, p0, Landroid/view/WindowManager$LayoutParams;->extraFlags:I
+ 
+     invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
+ 
      iget v0, p0, Landroid/view/WindowManager$LayoutParams;->privateFlags:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
