*** ./smali/com/android/server/connectivity/Tethering.smali	2014-08-20 05:49:50.964016153 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/services.jar.out/./smali/com/android/server/connectivity/Tethering.smali	2014-08-20 05:52:36.276020440 +0700
***************
*** 1213,1218 ****
--- 1213,1224 ----
      return-void
  
      :cond_1
+     invoke-static {}, Lcom/android/server/connectivity/Injector$TetheringHook;->isUsbShareNet()Z
+ 
+     move-result v0
+ 
+     if-nez v0, :cond_0
+ 
      iget-object v0, p0, Lcom/android/server/connectivity/Tethering;->mTetheredNotification:Landroid/app/Notification;
  
      if-eqz v0, :cond_2
***************
*** 1243,1252 ****
  
      invoke-virtual {v2, v0, v3}, Landroid/content/Intent;->setClassName(Ljava/lang/String;Ljava/lang/String;)Landroid/content/Intent;
  
-     const/high16 v0, 0x4000
- 
-     invoke-virtual {v2, v0}, Landroid/content/Intent;->setFlags(I)Landroid/content/Intent;
- 
      iget-object v0, p0, Lcom/android/server/connectivity/Tethering;->mContext:Landroid/content/Context;
  
      sget-object v5, Landroid/os/UserHandle;->CURRENT:Landroid/os/UserHandle;
--- 1249,1254 ----
