*** ./smali/com/android/server/am/ActivityStack.smali	2014-08-20 05:49:39.984015869 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/services.jar.out/./smali/com/android/server/am/ActivityStack.smali	2014-08-20 05:52:25.312020156 +0700
***************
*** 268,273 ****
--- 268,275 ----
  
      iput v0, p0, Lcom/android/server/am/ActivityStack;->mCurrentUser:I
  
+     invoke-static {p0, p1}, Lcom/android/server/am/Injector$ActivityStackHook;->after_ActivityStack(Lcom/android/server/am/ActivityStack;Lcom/android/server/am/ActivityManagerService;)V
+ 
      return-void
  .end method
  
***************
*** 2453,2459 ****
      check-cast v4, Lcom/android/server/am/ActivityRecord;
  
      .local v4, r:Lcom/android/server/am/ActivityRecord;
!     iget-boolean v6, v4, Lcom/android/server/am/ActivityRecord;->finishing:Z
  
      if-eqz v6, :cond_1
  
--- 2455,2463 ----
      check-cast v4, Lcom/android/server/am/ActivityRecord;
  
      .local v4, r:Lcom/android/server/am/ActivityRecord;
!     invoke-static {p3, v4}, Lcom/android/server/am/Injector$ActivityStackHook;->destroyHomeOrFinishing(Ljava/lang/String;Lcom/android/server/am/ActivityRecord;)Z
! 
!     move-result v6
  
      if-eqz v6, :cond_1
  
***************
*** 3624,3629 ****
--- 3628,3644 ----
      if-eqz v5, :cond_c
  
      :cond_7
+     invoke-static {p1, v0}, Lcom/android/server/am/Injector$ActivityStackHook;->needRestartActivity(Lcom/android/server/am/ActivityRecord;I)Z
+ 
+     move-result v5
+ 
+     if-eqz v5, :cond_miui_0
+ 
+     const/4 v5, 0x1
+ 
+     return v5
+ 
+     :cond_miui_0
      iget v5, p1, Lcom/android/server/am/ActivityRecord;->configChangeFlags:I
  
      or-int/2addr v5, v0
***************
*** 8453,8458 ****
--- 8468,8477 ----
  
      move-result v0
  
+     invoke-static {p0, v0}, Lcom/android/server/am/Injector$ActivityStackHook;->setForegroundProcess(Lcom/android/server/am/ActivityStack;Z)Z
+ 
+     move-result v0
+ 
      return v0
  .end method
  
***************
*** 10470,10475 ****
--- 10489,10496 ----
      .parameter "options"
  
      .prologue
+     invoke-static {}, Lcom/android/server/am/Injector$ActivityStackHook;->before_startActivityLocked()V
+ 
      move-object/from16 v0, p1
  
      iget-object v0, v0, Lcom/android/server/am/ActivityRecord;->task:Lcom/android/server/am/TaskRecord;
