*** ./smali/com/android/server/am/ActivityManagerService$2.smali	2014-08-20 05:49:40.960015894 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/services.jar.out/./smali/com/android/server/am/ActivityManagerService$2.smali	2014-08-20 05:52:26.320020182 +0700
***************
*** 284,289 ****
--- 284,301 ----
  
      if-nez v4, :cond_7
  
+     move-object/from16 v0, p0
+ 
+     iget-object v4, v0, Lcom/android/server/am/ActivityManagerService$2;->this$0:Lcom/android/server/am/ActivityManagerService;
+ 
+     move-object/from16 v0, v30
+ 
+     invoke-static {v4, v0}, Lcom/android/server/am/Injector$ActivityManagerServiceHook;->showAppCrashDialog(Lcom/android/server/am/ActivityManagerService;Ljava/util/HashMap;)Z
+ 
+     move-result v4
+ 
+     if-nez v4, :cond_6
+ 
      new-instance v8, Lcom/android/server/am/AppErrorDialog;
  
      move-object/from16 v0, p0
