*** ./smali/com/android/server/am/AppNotRespondingDialog.smali	2014-08-20 05:49:41.984015921 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/services.jar.out/./smali/com/android/server/am/AppNotRespondingDialog.smali	2014-08-20 05:52:27.320020208 +0700
***************
*** 320,325 ****
--- 320,334 ----
  
  
  # virtual methods
+ .method getProc()Lcom/android/server/am/ProcessRecord;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/server/am/AppNotRespondingDialog;->mProc:Lcom/android/server/am/ProcessRecord;
+ 
+     return-object v0
+ .end method
+ 
  .method public onStop()V
      .locals 0
  
