*** ./smali/com/android/internal/telephony/SMSDispatcher.smali	2014-08-20 05:50:05.364016527 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/telephony-common.jar.out/./smali/com/android/internal/telephony/SMSDispatcher.smali	2014-08-20 05:52:50.708020814 +0700
***************
*** 84,89 ****
--- 84,102 ----
  
  .field private mImsSMSDispatcher:Lcom/android/internal/telephony/ImsSMSDispatcher;
  
+ .field public mIsSynchronizedSending:Z
+ 
+ .field public mPendingMessagesList:Ljava/util/LinkedList;
+     .annotation system Ldalvik/annotation/Signature;
+         value = {
+             "Ljava/util/LinkedList",
+             "<",
+             "Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;",
+             ">;"
+         }
+     .end annotation
+ .end field
+ 
  .field private mPendingTrackerCount:I
  
  .field protected mPhone:Lcom/android/internal/telephony/PhoneBase;
***************
*** 100,105 ****
--- 113,120 ----
  
  .field protected mSmsSendDisabled:Z
  
+ .field public mSyncronousSending:Z
+ 
  .field protected final mTelephonyManager:Landroid/telephony/TelephonyManager;
  
  .field private mUsageMonitor:Lcom/android/internal/telephony/SmsUsageMonitor;
***************
*** 150,155 ****
--- 165,172 ----
  
      iput v0, p0, Lcom/android/internal/telephony/SMSDispatcher;->mRemainingMessages:I
  
+     iput-boolean v2, p0, Lcom/android/internal/telephony/SMSDispatcher;->mIsSynchronizedSending:Z
+ 
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0}, Ljava/util/ArrayList;-><init>()V
***************
*** 247,252 ****
--- 264,273 ----
      :goto_0
      iput-boolean v0, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSmsSendDisabled:Z
  
+     iput-boolean v2, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSyncronousSending:Z
+ 
+     invoke-static {p0}, Lcom/android/internal/telephony/Injector$SMSDispatcherHook;->initSynchronousSending(Lcom/android/internal/telephony/SMSDispatcher;)V
+ 
      const-string v0, "SMSDispatcher"
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 1676,1681 ****
--- 1697,1706 ----
  
      const/4 v10, -0x1
  
+     const/4 v7, 0x0
+ 
+     iput-boolean v7, p0, Lcom/android/internal/telephony/SMSDispatcher;->mIsSynchronizedSending:Z
+ 
      iget-object v6, p1, Landroid/os/AsyncResult;->userObj:Ljava/lang/Object;
  
      check-cast v6, Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;
***************
*** 1769,1774 ****
--- 1794,1802 ----
      .end local v3           #sendNext:Landroid/content/Intent;
      :cond_3
      :goto_1
+     invoke-static {p0}, Lcom/android/internal/telephony/Injector$SMSDispatcherHook;->processNextPendingMessageIfSynchronous(Lcom/android/internal/telephony/SMSDispatcher;)V
+ 
+     :cond_miui
      return-void
  
      :cond_4
***************
*** 1952,1958 ****
      :cond_9
      iget-object v7, v6, Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;->mSentIntent:Landroid/app/PendingIntent;
  
!     if-eqz v7, :cond_3
  
      const/4 v0, 0x1
  
--- 1980,1986 ----
      :cond_9
      iget-object v7, v6, Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;->mSentIntent:Landroid/app/PendingIntent;
  
!     if-eqz v7, :cond_miui
  
      const/4 v0, 0x1
  
***************
*** 2558,2564 ****
      goto/16 :goto_1
  
      :cond_7
!     invoke-virtual {p0, p1}, Lcom/android/internal/telephony/SMSDispatcher;->sendSms(Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;)V
  
      goto/16 :goto_1
  
--- 2586,2592 ----
      goto/16 :goto_1
  
      :cond_7
!     invoke-static {p0, p1}, Lcom/android/internal/telephony/Injector$SMSDispatcherHook;->enqueueOrSendSms(Lcom/android/internal/telephony/SMSDispatcher;Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;)V
  
      goto/16 :goto_1
  
