*** ./smali/android/app/ActivityOptions.smali	2014-08-20 05:48:08.420013494 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/framework.jar.out/./smali/android/app/ActivityOptions.smali	2014-08-20 05:50:53.288017770 +0700
***************
*** 306,311 ****
--- 306,327 ----
      return-object v0
  .end method
  
+ .method public static makeDelayedThumbnailScaleUpAnimation(Landroid/view/View;Landroid/graphics/Bitmap;IILandroid/app/ActivityOptions$OnAnimationStartedListener;)Landroid/app/ActivityOptions;
+     .locals 1
+     .parameter "source"
+     .parameter "thumbnail"
+     .parameter "startX"
+     .parameter "startY"
+     .parameter "listener"
+ 
+     .prologue
+     invoke-static {p0, p1, p2, p3, p4}, Landroid/app/ActivityOptions;->makeThumbnailScaleUpAnimation(Landroid/view/View;Landroid/graphics/Bitmap;IILandroid/app/ActivityOptions$OnAnimationStartedListener;)Landroid/app/ActivityOptions;
+ 
+     move-result-object v0
+ 
+     return-object v0
+ .end method
+ 
  .method public static makeScaleUpAnimation(Landroid/view/View;IIII)Landroid/app/ActivityOptions;
      .locals 4
      .parameter "source"
