*** ./smali/android/app/Dialog.smali	2014-08-20 05:48:13.384013623 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/framework.jar.out/./smali/android/app/Dialog.smali	2014-08-20 05:50:58.240017898 +0700
***************
*** 197,202 ****
--- 197,204 ----
  
      iput-object v2, p0, Landroid/app/Dialog;->mListenersHandler:Landroid/os/Handler;
  
+     invoke-static {p0}, Landroid/app/Injector$DialogHook;->after_Dialog(Landroid/app/Dialog;)V
+ 
      return-void
  
      .end local v1           #w:Landroid/view/Window;
***************
*** 477,482 ****
--- 479,486 ----
      iget-object v1, p0, Landroid/app/Dialog;->mDecor:Landroid/view/View;
  
      invoke-interface {v0, v1}, Landroid/view/WindowManager;->removeViewImmediate(Landroid/view/View;)V
+ 
+     invoke-static {p0}, Landroid/app/Injector$DialogHook;->onWindowHide(Landroid/app/Dialog;)V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
***************
*** 845,850 ****
--- 849,880 ----
      return-object v0
  .end method
  
+ .method public getMiuiActionBar()Lmiui/v5/app/MiuiActionBar;
+     .locals 2
+ 
+     .prologue
+     invoke-virtual {p0}, Landroid/app/Dialog;->getActionBar()Landroid/app/ActionBar;
+ 
+     move-result-object v0
+ 
+     .local v0, bar:Landroid/app/ActionBar;
+     instance-of v1, v0, Lmiui/v5/app/MiuiActionBar;
+ 
+     if-eqz v1, :cond_0
+ 
+     check-cast v0, Lmiui/v5/app/MiuiActionBar;
+ 
+     .end local v0           #bar:Landroid/app/ActionBar;
+     :goto_0
+     return-object v0
+ 
+     .restart local v0       #bar:Landroid/app/ActionBar;
+     :cond_0
+     const/4 v0, 0x0
+ 
+     goto :goto_0
+ .end method
+ 
  .method public final getOwnerActivity()Landroid/app/Activity;
      .locals 1
  
***************
*** 2057,2065 ****
  
      invoke-virtual {v3, v4}, Landroid/view/Window;->setDefaultLogo(I)V
  
!     new-instance v3, Lcom/android/internal/app/ActionBarImpl;
  
!     invoke-direct {v3, p0}, Lcom/android/internal/app/ActionBarImpl;-><init>(Landroid/app/Dialog;)V
  
      iput-object v3, p0, Landroid/app/Dialog;->mActionBar:Lcom/android/internal/app/ActionBarImpl;
  
--- 2087,2095 ----
  
      invoke-virtual {v3, v4}, Landroid/view/Window;->setDefaultLogo(I)V
  
!     invoke-static {p0}, Landroid/app/Injector$DialogHook;->createActionBarImpl(Landroid/app/Dialog;)Lcom/android/internal/app/ActionBarImpl;
  
!     move-result-object v3
  
      iput-object v3, p0, Landroid/app/Dialog;->mActionBar:Lcom/android/internal/app/ActionBarImpl;
  
***************
*** 2107,2112 ****
--- 2137,2144 ----
      iput-boolean v3, p0, Landroid/app/Dialog;->mShowing:Z
  
      invoke-direct {p0}, Landroid/app/Dialog;->sendShowMessage()V
+ 
+     invoke-static {p0}, Landroid/app/Injector$DialogHook;->after_show(Landroid/app/Dialog;)V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
