*** ./smali/android/app/Notification.smali	2014-08-20 05:48:10.416013546 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/framework.jar.out/./smali/android/app/Notification.smali	2014-08-20 05:50:55.268017821 +0700
***************
*** 139,144 ****
--- 139,146 ----
  
  .field public deleteIntent:Landroid/app/PendingIntent;
  
+ .field public extraNotification:Lmiui/app/ExtraNotification;
+ 
  .field public extras:Landroid/os/Bundle;
  
  .field public flags:I
***************
*** 194,199 ****
--- 196,207 ----
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     new-instance v0, Lmiui/app/ExtraNotification;
+ 
+     invoke-direct {v0}, Lmiui/app/ExtraNotification;-><init>()V
+ 
+     iput-object v0, p0, Landroid/app/Notification;->extraNotification:Lmiui/app/ExtraNotification;
+ 
      const/4 v0, -0x1
  
      iput v0, p0, Landroid/app/Notification;->audioStreamType:I
***************
*** 228,233 ****
--- 236,247 ----
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     new-instance v0, Lmiui/app/ExtraNotification;
+ 
+     invoke-direct {v0}, Lmiui/app/ExtraNotification;-><init>()V
+ 
+     iput-object v0, p0, Landroid/app/Notification;->extraNotification:Lmiui/app/ExtraNotification;
+ 
      const/4 v0, -0x1
  
      iput v0, p0, Landroid/app/Notification;->audioStreamType:I
***************
*** 262,267 ****
--- 276,287 ----
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     new-instance v0, Lmiui/app/ExtraNotification;
+ 
+     invoke-direct {v0}, Lmiui/app/ExtraNotification;-><init>()V
+ 
+     iput-object v0, p0, Landroid/app/Notification;->extraNotification:Lmiui/app/ExtraNotification;
+ 
      const/4 v0, -0x1
  
      iput v0, p0, Landroid/app/Notification;->audioStreamType:I
***************
*** 294,299 ****
--- 314,325 ----
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     new-instance v1, Lmiui/app/ExtraNotification;
+ 
+     invoke-direct {v1}, Lmiui/app/ExtraNotification;-><init>()V
+ 
+     iput-object v1, p0, Landroid/app/Notification;->extraNotification:Lmiui/app/ExtraNotification;
+ 
      const/4 v1, -0x1
  
      iput v1, p0, Landroid/app/Notification;->audioStreamType:I
***************
*** 556,561 ****
--- 582,591 ----
      iput-object v1, p0, Landroid/app/Notification;->bigContentView:Landroid/widget/RemoteViews;
  
      :cond_8
+     iget-object v1, p0, Landroid/app/Notification;->extraNotification:Lmiui/app/ExtraNotification;
+ 
+     invoke-virtual {v1, p1}, Lmiui/app/ExtraNotification;->readFromParcel(Landroid/os/Parcel;)V
+ 
      return-void
  .end method
  
***************
*** 628,633 ****
--- 658,665 ----
  
      invoke-virtual {p0, v0, v1}, Landroid/app/Notification;->cloneInto(Landroid/app/Notification;Z)V
  
+     invoke-static {p0, v0}, Landroid/app/Injector$NotificationHook;->after_clone(Landroid/app/Notification;Landroid/app/Notification;)V
+ 
      return-object v0
  .end method
  
***************
*** 1546,1551 ****
--- 1578,1585 ----
      invoke-virtual {v0, p1, v2}, Landroid/widget/RemoteViews;->writeToParcel(Landroid/os/Parcel;I)V
  
      :goto_8
+     invoke-static {p0, p1, p2}, Landroid/app/Injector$NotificationHook;->after_writeToParcel(Landroid/app/Notification;Landroid/os/Parcel;I)V
+ 
      return-void
  
      :cond_0
