*** ./smali/android/app/ResourcesManager.smali	2014-08-20 05:48:12.752013607 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/framework.jar.out/./smali/android/app/ResourcesManager.smali	2014-08-20 05:50:57.608017882 +0700
***************
*** 215,220 ****
--- 215,222 ----
      or-int/lit16 v0, v0, 0xd00
  
      :cond_3
+     invoke-static {v0, p1}, Landroid/app/Injector$ActivityThreadHook;->handleExtraConfigurationChanges(ILandroid/content/res/Configuration;)V
+ 
      iget-object v10, p1, Landroid/content/res/Configuration;->locale:Ljava/util/Locale;
  
      if-eqz v10, :cond_4
***************
*** 772,785 ****
  
      :cond_5
      :goto_3
!     new-instance v7, Landroid/content/res/Resources;
  
!     .end local v7           #r:Landroid/content/res/Resources;
!     move-object/from16 v11, p4
  
!     move-object/from16 v12, p5
  
!     invoke-direct/range {v7 .. v12}, Landroid/content/res/Resources;-><init>(Landroid/content/res/AssetManager;Landroid/util/DisplayMetrics;Landroid/content/res/Configuration;Landroid/content/res/CompatibilityInfo;Landroid/os/IBinder;)V
  
      .restart local v7       #r:Landroid/content/res/Resources;
      monitor-enter p0
--- 774,786 ----
  
      :cond_5
      :goto_3
!     move-object/from16 v0, p4
  
!     move-object/from16 v1, p5
  
!     invoke-static {v8, v9, v10, v0, v1}, Landroid/app/Injector$ActivityThreadHook;->createResources(Landroid/content/res/AssetManager;Landroid/util/DisplayMetrics;Landroid/content/res/Configuration;Landroid/content/res/CompatibilityInfo;Landroid/os/IBinder;)Landroid/content/res/Resources;
  
!     move-result-object v7
  
      .restart local v7       #r:Landroid/content/res/Resources;
      monitor-enter p0
