*** ./smali/android/app/SearchDialog.smali	2014-08-20 05:48:09.724013528 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/framework.jar.out/./smali/android/app/SearchDialog.smali	2014-08-20 05:50:54.584017803 +0700
***************
*** 776,781 ****
--- 776,790 ----
      .prologue
      const/4 v7, 0x0
  
+     invoke-static {p0}, Landroid/app/Injector$SearchDialogHook;->before_updateSearchAppIcon(Landroid/app/SearchDialog;)Z
+ 
+     move-result v4
+ 
+     if-eqz v4, :cond_0
+ 
+     return-void
+ 
+     :cond_0
      invoke-virtual {p0}, Landroid/app/SearchDialog;->getContext()Landroid/content/Context;
  
      move-result-object v4
***************
*** 1085,1090 ****
--- 1094,1117 ----
  
  
  # virtual methods
+ .method getAppIcon()Landroid/widget/ImageView;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Landroid/app/SearchDialog;->mAppIcon:Landroid/widget/ImageView;
+ 
+     return-object v0
+ .end method
+ 
+ .method getCloseSearch()Landroid/view/View;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Landroid/app/SearchDialog;->mCloseSearch:Landroid/view/View;
+ 
+     return-object v0
+ .end method
+ 
  .method public hide()V
      .locals 3
  
