*** ./smali/android/widget/AbsListView$FlingRunnable.smali	2014-08-20 05:48:26.904013974 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/framework.jar.out/./smali/android/widget/AbsListView$FlingRunnable.smali	2014-08-20 05:51:11.928018253 +0700
***************
*** 70,75 ****
--- 70,125 ----
      return-object v0
  .end method
  
+ .method private endFling2()V
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Landroid/widget/AbsListView$FlingRunnable;->this$0:Landroid/widget/AbsListView;
+ 
+     invoke-virtual {v0}, Landroid/widget/AbsListView;->getScrollY()I
+ 
+     move-result v0
+ 
+     if-eqz v0, :cond_0
+ 
+     invoke-virtual {p0}, Landroid/widget/AbsListView$FlingRunnable;->startSpringback()V
+ 
+     :goto_0
+     return-void
+ 
+     :cond_0
+     invoke-virtual {p0}, Landroid/widget/AbsListView$FlingRunnable;->endFling()V
+ 
+     goto :goto_0
+ .end method
+ 
+ .method private scaleOverScrollDeltaY(II)I
+     .locals 2
+     .parameter "deltaY"
+     .parameter "scrollY"
+ 
+     .prologue
+     move v0, p1
+ 
+     .local v0, retval:I
+     iget-object v1, p0, Landroid/widget/AbsListView$FlingRunnable;->mScroller:Landroid/widget/OverScroller;
+ 
+     invoke-virtual {v1}, Landroid/widget/OverScroller;->checkSpringBackState()Z
+ 
+     move-result v1
+ 
+     if-nez v1, :cond_0
+ 
+     iget-object v1, p0, Landroid/widget/AbsListView$FlingRunnable;->this$0:Landroid/widget/AbsListView;
+ 
+     invoke-static {v1, p1, p2}, Landroid/widget/Injector$AbsListViewHook;->scaleOverScrollDeltaY(Landroid/widget/AbsListView;II)I
+ 
+     move-result v0
+ 
+     :cond_0
+     return v0
+ .end method
+ 
  
  # virtual methods
  .method edgeReached(I)V
***************
*** 271,276 ****
--- 321,335 ----
      return-void
  .end method
  
+ .method protected getScroller()Landroid/widget/OverScroller;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Landroid/widget/AbsListView$FlingRunnable;->mScroller:Landroid/widget/OverScroller;
+ 
+     return-object v0
+ .end method
+ 
  .method public run()V
      .locals 30
  
***************
*** 513,518 ****
--- 572,589 ----
  
      iget-object v2, v0, Landroid/widget/AbsListView$FlingRunnable;->this$0:Landroid/widget/AbsListView;
  
+     invoke-virtual {v2}, Landroid/widget/AbsListView;->getScrollY()I
+ 
+     move-result v3
+ 
+     invoke-static {v2, v4, v3}, Landroid/widget/Injector$AbsListViewHook;->scaleOverScrollDeltaY(Landroid/widget/AbsListView;II)I
+ 
+     move-result v4
+ 
+     move-object/from16 v0, p0
+ 
+     iget-object v2, v0, Landroid/widget/AbsListView$FlingRunnable;->this$0:Landroid/widget/AbsListView;
+ 
      const/4 v3, 0x0
  
      const/4 v5, 0x0
***************
*** 731,737 ****
      move-result v19
  
      .local v19, currY:I
!     sub-int v7, v19, v9
  
      .local v7, deltaY:I
      move-object/from16 v0, p0
--- 802,814 ----
      move-result v19
  
      .local v19, currY:I
!     sub-int v2, v19, v9
! 
!     move-object/from16 v0, p0
! 
!     invoke-direct {v0, v2, v9}, Landroid/widget/AbsListView$FlingRunnable;->scaleOverScrollDeltaY(II)I
! 
!     move-result v7
  
      .local v7, deltaY:I
      move-object/from16 v0, p0
***************
*** 855,861 ****
      .end local v9           #scrollY:I
      .end local v19           #currY:I
      :cond_11
!     invoke-virtual/range {p0 .. p0}, Landroid/widget/AbsListView$FlingRunnable;->endFling()V
  
      goto/16 :goto_0
  
--- 932,938 ----
      .end local v9           #scrollY:I
      .end local v19           #currY:I
      :cond_11
!     invoke-direct/range {p0 .. p0}, Landroid/widget/AbsListView$FlingRunnable;->endFling2()V
  
      goto/16 :goto_0
  
