*** ./smali/android/widget/QuickContactBadge.smali	2014-08-20 05:48:24.672013916 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/framework.jar.out/./smali/android/widget/QuickContactBadge.smali	2014-08-20 05:51:09.644018194 +0700
***************
*** 594,603 ****
--- 594,619 ----
  
      move-result v2
  
+     iget v3, p0, Landroid/widget/QuickContactBadge;->mPaddingLeft:I
+ 
+     sub-int/2addr v2, v3
+ 
+     iget v3, p0, Landroid/widget/QuickContactBadge;->mPaddingRight:I
+ 
+     sub-int/2addr v2, v3
+ 
      invoke-virtual {p0}, Landroid/widget/QuickContactBadge;->getHeight()I
  
      move-result v3
  
+     iget v0, p0, Landroid/widget/QuickContactBadge;->mPaddingTop:I
+ 
+     sub-int/2addr v3, v0
+ 
+     iget v0, p0, Landroid/widget/QuickContactBadge;->mPaddingBottom:I
+ 
+     sub-int/2addr v3, v0
+ 
      invoke-virtual {v1, v4, v4, v2, v3}, Landroid/graphics/drawable/Drawable;->setBounds(IIII)V
  
      iget v1, p0, Landroid/widget/QuickContactBadge;->mPaddingTop:I
