*** ./smali/com/android/server/wm/WindowManagerService.smali	2014-08-20 05:49:45.072016001 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/services.jar.out/./smali/com/android/server/wm/WindowManagerService.smali	2014-08-20 05:52:30.404020288 +0700
***************
*** 17514,17519 ****
--- 17514,17530 ----
      :cond_2
      iput-boolean v5, p0, Lcom/android/server/wm/WindowManagerService;->mSafeMode:Z
  
+     const/4 v6, 0x0
+ 
+     if-lez v1, :cond_miui_0
+ 
+     if-lez v4, :cond_miui_0
+ 
+     const/4 v6, 0x1
+ 
+     :cond_miui_0
+     iput-boolean v6, p0, Lcom/android/server/wm/WindowManagerService;->mSafeMode:Z
+ 
      :try_start_0
      const-string v5, "persist.sys.safemode"
  
***************
*** 26871,26876 ****
--- 26882,26898 ----
      goto :goto_1
  .end method
  
+ .method public rebootSafeMode()V
+     .locals 1
+ 
+     .prologue
+     const/4 v0, 0x0
+ 
+     invoke-virtual {p0, v0}, Lcom/android/server/wm/WindowManagerService;->rebootSafeMode(Z)V
+ 
+     return-void
+ .end method
+ 
  .method public rebootSafeMode(Z)V
      .locals 1
      .parameter "confirm"
***************
*** 36003,36008 ****
--- 36025,36041 ----
      goto :goto_1
  .end method
  
+ .method public shutdown()V
+     .locals 1
+ 
+     .prologue
+     const/4 v0, 0x0
+ 
+     invoke-virtual {p0, v0}, Lcom/android/server/wm/WindowManagerService;->shutdown(Z)V
+ 
+     return-void
+ .end method
+ 
  .method public shutdown(Z)V
      .locals 1
      .parameter "confirm"
