*** ./smali/android/content/res/Resources.smali	2014-08-20 05:47:31.980012550 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/framework.jar.out/./smali/android/content/res/Resources.smali	2014-08-20 05:50:16.696016821 +0700
***************
*** 217,223 ****
      return-void
  .end method
  
! .method private constructor <init>()V
      .locals 4
  
      .prologue
--- 217,223 ----
      return-void
  .end method
  
! .method constructor <init>()V
      .locals 4
  
      .prologue
***************
*** 611,616 ****
--- 611,647 ----
      return-void
  .end method
  
+ .method static clearPreloadedCache()V
+     .locals 2
+ 
+     .prologue
+     sget-object v0, Landroid/content/res/Resources;->sPreloadedDrawables:[Landroid/util/LongSparseArray;
+ 
+     const/4 v1, 0x0
+ 
+     aget-object v0, v0, v1
+ 
+     invoke-virtual {v0}, Landroid/util/LongSparseArray;->clear()V
+ 
+     sget-object v0, Landroid/content/res/Resources;->sPreloadedDrawables:[Landroid/util/LongSparseArray;
+ 
+     const/4 v1, 0x1
+ 
+     aget-object v0, v0, v1
+ 
+     invoke-virtual {v0}, Landroid/util/LongSparseArray;->clear()V
+ 
+     sget-object v0, Landroid/content/res/Resources;->sPreloadedColorStateLists:Landroid/util/LongSparseArray;
+ 
+     invoke-virtual {v0}, Landroid/util/LongSparseArray;->clear()V
+ 
+     sget-object v0, Landroid/content/res/Resources;->sPreloadedColorDrawables:Landroid/util/LongSparseArray;
+ 
+     invoke-virtual {v0}, Landroid/util/LongSparseArray;->clear()V
+ 
+     return-void
+ .end method
+ 
  .method private getCachedColorStateList(J)Landroid/content/res/ColorStateList;
      .locals 4
      .parameter "key"
***************
*** 826,832 ****
  
      new-array v4, v4, [I
  
!     invoke-direct {v0, p0, v2, v4, p1}, Landroid/content/res/TypedArray;-><init>(Landroid/content/res/Resources;[I[II)V
  
      monitor-exit v3
      :try_end_1
--- 857,865 ----
  
      new-array v4, v4, [I
  
!     invoke-static {p0, v2, v4, p1}, Landroid/content/res/Injector$ResourcesHook;->createTypedArray(Landroid/content/res/Resources;[I[II)Landroid/content/res/TypedArray;
! 
!     move-result-object v0
  
      monitor-exit v3
      :try_end_1
***************
*** 889,902 ****
      .local v0, ret:Landroid/content/res/Resources;
      if-nez v0, :cond_0
  
!     new-instance v0, Landroid/content/res/Resources;
  
!     .end local v0           #ret:Landroid/content/res/Resources;
!     invoke-direct {v0}, Landroid/content/res/Resources;-><init>()V
  
-     .restart local v0       #ret:Landroid/content/res/Resources;
      sput-object v0, Landroid/content/res/Resources;->mSystem:Landroid/content/res/Resources;
  
      :cond_0
      monitor-exit v2
  
--- 922,935 ----
      .local v0, ret:Landroid/content/res/Resources;
      if-nez v0, :cond_0
  
!     invoke-static {}, Landroid/content/res/Injector$ResourcesHook;->createResources()Landroid/content/res/Resources;
  
!     move-result-object v0
  
      sput-object v0, Landroid/content/res/Resources;->mSystem:Landroid/content/res/Resources;
  
+     invoke-static {v0}, Landroid/content/res/Injector$ResourcesHook;->initResources(Landroid/content/res/Resources;)V
+ 
      :cond_0
      monitor-exit v2
  
***************
*** 2610,2615 ****
--- 2643,2685 ----
      goto :goto_0
  .end method
  
+ .method getPreloadedDrawable(JI)Landroid/graphics/drawable/Drawable$ConstantState;
+     .locals 3
+     .parameter "key"
+     .parameter "id"
+ 
+     .prologue
+     sget-object v1, Landroid/content/res/Resources;->sPreloadedDrawables:[Landroid/util/LongSparseArray;
+ 
+     iget-object v2, p0, Landroid/content/res/Resources;->mConfiguration:Landroid/content/res/Configuration;
+ 
+     invoke-virtual {v2}, Landroid/content/res/Configuration;->getLayoutDirection()I
+ 
+     move-result v2
+ 
+     aget-object v1, v1, v2
+ 
+     invoke-virtual {v1, p1, p2}, Landroid/util/LongSparseArray;->get(J)Ljava/lang/Object;
+ 
+     move-result-object v0
+ 
+     check-cast v0, Landroid/graphics/drawable/Drawable$ConstantState;
+ 
+     .local v0, cs:Landroid/graphics/drawable/Drawable$ConstantState;
+     if-eqz v0, :cond_0
+ 
+     invoke-virtual {p0, p3}, Landroid/content/res/Resources;->isPreloadOverlayed(I)Z
+ 
+     move-result v1
+ 
+     if-eqz v1, :cond_0
+ 
+     const/4 v0, 0x0
+ 
+     :cond_0
+     return-object v0
+ .end method
+ 
  .method public getPreloadedDrawables()Landroid/util/LongSparseArray;
      .locals 2
      .annotation system Ldalvik/annotation/Signature;
***************
*** 3457,3462 ****
--- 3527,3551 ----
      return-object v0
  .end method
  
+ .method isPreloadOverlayed(I)Z
+     .locals 1
+     .parameter "id"
+ 
+     .prologue
+     const/4 v0, 0x0
+ 
+     return v0
+ .end method
+ 
+ .method isPreloading()Z
+     .locals 1
+ 
+     .prologue
+     iget-boolean v0, p0, Landroid/content/res/Resources;->mPreloading:Z
+ 
+     return v0
+ .end method
+ 
  .method loadColorStateList(Landroid/util/TypedValue;I)Landroid/content/res/ColorStateList;
      .locals 12
      .parameter "value"
***************
*** 4053,4080 ****
      .end local v7           #dr:Landroid/graphics/drawable/Drawable;
      .restart local v6       #dr:Landroid/graphics/drawable/Drawable;
      :cond_6
-     sget-object v17, Landroid/content/res/Resources;->sPreloadedDrawables:[Landroid/util/LongSparseArray;
- 
      move-object/from16 v0, p0
  
!     iget-object v0, v0, Landroid/content/res/Resources;->mConfiguration:Landroid/content/res/Configuration;
! 
!     move-object/from16 v18, v0
! 
!     invoke-virtual/range {v18 .. v18}, Landroid/content/res/Configuration;->getLayoutDirection()I
! 
!     move-result v18
! 
!     aget-object v17, v17, v18
! 
!     move-object/from16 v0, v17
  
!     invoke-virtual {v0, v12, v13}, Landroid/util/LongSparseArray;->get(J)Ljava/lang/Object;
  
      move-result-object v5
  
-     check-cast v5, Landroid/graphics/drawable/Drawable$ConstantState;
- 
      .restart local v5       #cs:Landroid/graphics/drawable/Drawable$ConstantState;
      goto :goto_3
  
--- 4142,4155 ----
      .end local v7           #dr:Landroid/graphics/drawable/Drawable;
      .restart local v6       #dr:Landroid/graphics/drawable/Drawable;
      :cond_6
      move-object/from16 v0, p0
  
!     move/from16 v1, p2
  
!     invoke-virtual {v0, v12, v13, v1}, Landroid/content/res/Resources;->getPreloadedDrawable(JI)Landroid/graphics/drawable/Drawable$ConstantState;
  
      move-result-object v5
  
      .restart local v5       #cs:Landroid/graphics/drawable/Drawable$ConstantState;
      goto :goto_3
  
***************
*** 4297,4305 ****
  
      move-object/from16 v1, p1
  
!     move-object/from16 v2, v17
  
!     invoke-static {v0, v1, v10, v9, v2}, Landroid/graphics/drawable/Drawable;->createFromResourceStream(Landroid/content/res/Resources;Landroid/util/TypedValue;Ljava/io/InputStream;Ljava/lang/String;Landroid/graphics/BitmapFactory$Options;)Landroid/graphics/drawable/Drawable;
  
      move-result-object v6
  
--- 4372,4380 ----
  
      move-object/from16 v1, p1
  
!     move/from16 v2, p2
  
!     invoke-static {v0, v1, v10, v9, v2}, Landroid/content/res/Injector$ResourcesHook;->createFromResourceStream(Landroid/content/res/Resources;Landroid/util/TypedValue;Ljava/io/InputStream;Ljava/lang/String;I)Landroid/graphics/drawable/Drawable;
  
      move-result-object v6
  
***************
*** 4515,4520 ****
--- 4590,4614 ----
      goto :goto_6
  .end method
  
+ .method loadOverlayDrawable(Landroid/util/TypedValue;I)Landroid/graphics/drawable/Drawable;
+     .locals 1
+     .parameter "value"
+     .parameter "id"
+ 
+     .prologue
+     const/4 v0, 0x0
+ 
+     return-object v0
+ .end method
+ 
+ .method loadOverlayTypedArray(Landroid/content/res/TypedArray;)Landroid/content/res/TypedArray;
+     .locals 0
+     .parameter "array"
+ 
+     .prologue
+     return-object p1
+ .end method
+ 
  .method loadXmlResourceParser(ILjava/lang/String;)Landroid/content/res/XmlResourceParser;
      .locals 5
      .parameter "id"
***************
*** 4919,4925 ****
  
      iput-object v2, v0, Landroid/content/res/TypedArray;->mXml:Landroid/content/res/XmlBlock$Parser;
  
!     return-object v0
  .end method
  
  .method public obtainTypedArray(I)Landroid/content/res/TypedArray;
--- 5013,5023 ----
  
      iput-object v2, v0, Landroid/content/res/TypedArray;->mXml:Landroid/content/res/XmlBlock$Parser;
  
!     invoke-virtual {p0, v0}, Landroid/content/res/Resources;->loadOverlayTypedArray(Landroid/content/res/TypedArray;)Landroid/content/res/TypedArray;
! 
!     move-result-object v3
! 
!     return-object v3
  .end method
  
  .method public obtainTypedArray(I)Landroid/content/res/TypedArray;
***************
*** 4991,4997 ****
  
      aput v4, v2, v4
  
!     return-object v0
  .end method
  
  .method public openRawResource(I)Ljava/io/InputStream;
--- 5089,5099 ----
  
      aput v4, v2, v4
  
!     invoke-virtual {p0, v0}, Landroid/content/res/Resources;->loadOverlayTypedArray(Landroid/content/res/TypedArray;)Landroid/content/res/TypedArray;
! 
!     move-result-object v2
! 
!     return-object v2
  .end method
  
  .method public openRawResource(I)Ljava/io/InputStream;
***************
*** 5896,5902 ****
  
      invoke-static/range {v20 .. v20}, Landroid/content/pm/ActivityInfo;->activityInfoConfigToNative(I)I
  
!     move-result v20
  
      .end local v21           #density:I
      :cond_4
--- 5998,6010 ----
  
      invoke-static/range {v20 .. v20}, Landroid/content/pm/ActivityInfo;->activityInfoConfigToNative(I)I
  
!     move-result v2
! 
!     const/high16 v3, -0x8000
! 
!     and-int v3, v3, v20
! 
!     or-int v20, v2, v3
  
      .end local v21           #density:I
      :cond_4
