*** ./smali/com/android/internal/statusbar/IStatusBarService$Stub.smali	2014-08-20 05:49:28.092015560 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/framework2.jar.out/./smali/com/android/internal/statusbar/IStatusBarService$Stub.smali	2014-08-20 05:52:13.276019844 +0700
***************
*** 62,67 ****
--- 62,69 ----
  
  .field static final TRANSACTION_setImeWindowStatus:I = 0x8
  
+ .field static final TRANSACTION_setStatus:I = 0x2a
+ 
  .field static final TRANSACTION_setSystemUiVisibility:I = 0x11
  
  .field static final TRANSACTION_setWindowState:I = 0x16
***************
*** 830,835 ****
--- 832,894 ----
  
      goto/16 :goto_0
  
+     .end local v2           #_arg0:I
+     .end local v3           #_arg1:I
+     :sswitch_2a
+     const-string v1, "com.android.internal.statusbar.IStatusBarService"
+ 
+     move-object/from16 v0, p2
+ 
+     invoke-virtual {v0, v1}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
+ 
+     invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
+ 
+     move-result v2
+ 
+     .restart local v2       #_arg0:I
+     invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readStrongBinder()Landroid/os/IBinder;
+ 
+     move-result-object v3
+ 
+     .local v3, _arg1:Landroid/os/IBinder;
+     invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
+ 
+     move-result-object v4
+ 
+     .local v4, _arg2:Ljava/lang/String;
+     invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
+ 
+     move-result v1
+ 
+     if-eqz v1, :cond_miui
+ 
+     sget-object v1, Landroid/os/Bundle;->CREATOR:Landroid/os/Parcelable$Creator;
+ 
+     move-object/from16 v0, p2
+ 
+     invoke-interface {v1, v0}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
+ 
+     move-result-object v5
+ 
+     check-cast v5, Landroid/os/Bundle;
+ 
+     .local v5, _arg3:Landroid/os/Bundle;
+     :goto_miui
+     invoke-virtual {p0, v2, v3, v4, v5}, Lcom/android/internal/statusbar/IStatusBarService$Stub;->setStatus(ILandroid/os/IBinder;Ljava/lang/String;Landroid/os/Bundle;)V
+ 
+     invoke-virtual/range {p3 .. p3}, Landroid/os/Parcel;->writeNoException()V
+ 
+     const/4 v1, 0x1
+ 
+     goto/16 :goto_0
+ 
+     .end local v5           #_arg3:Landroid/os/Bundle;
+     :cond_miui
+     const/4 v5, 0x0
+ 
+     .restart local v5       #_arg3:Landroid/os/Bundle;
+     goto :goto_miui
+ 
      :sswitch_data_0
      .sparse-switch
          0x1 -> :sswitch_1
***************
*** 854,859 ****
--- 913,919 ----
          0x14 -> :sswitch_14
          0x15 -> :sswitch_15
          0x16 -> :sswitch_16
+         0x2a -> :sswitch_2a
          0x5f4e5446 -> :sswitch_0
      .end sparse-switch
  .end method
