*** ./smali/com/android/internal/statusbar/IStatusBarService$Stub$Proxy.smali	2014-08-20 05:49:28.176015563 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/framework2.jar.out/./smali/com/android/internal/statusbar/IStatusBarService$Stub$Proxy.smali	2014-08-20 05:52:13.364019846 +0700
***************
*** 1167,1172 ****
--- 1167,1255 ----
      throw v2
  .end method
  
+ .method public setStatus(ILandroid/os/IBinder;Ljava/lang/String;Landroid/os/Bundle;)V
+     .locals 5
+     .parameter "what"
+     .parameter "token"
+     .parameter "action"
+     .parameter "ext"
+     .annotation system Ldalvik/annotation/Throws;
+         value = {
+             Landroid/os/RemoteException;
+         }
+     .end annotation
+ 
+     .prologue
+     invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
+ 
+     move-result-object v0
+ 
+     .local v0, _data:Landroid/os/Parcel;
+     invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
+ 
+     move-result-object v1
+ 
+     .local v1, _reply:Landroid/os/Parcel;
+     :try_start_0
+     const-string v2, "com.android.internal.statusbar.IStatusBarService"
+ 
+     invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
+ 
+     invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeInt(I)V
+ 
+     invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
+ 
+     invoke-virtual {v0, p3}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
+ 
+     if-eqz p4, :cond_0
+ 
+     const/4 v2, 0x1
+ 
+     invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
+ 
+     const/4 v2, 0x0
+ 
+     invoke-virtual {p4, v0, v2}, Landroid/os/Bundle;->writeToParcel(Landroid/os/Parcel;I)V
+ 
+     :goto_0
+     iget-object v2, p0, Lcom/android/internal/statusbar/IStatusBarService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
+ 
+     const/16 v3, 0x2a
+ 
+     const/4 v4, 0x0
+ 
+     invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
+ 
+     invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
+     :try_end_0
+     .catchall {:try_start_0 .. :try_end_0} :catchall_0
+ 
+     invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
+ 
+     invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
+ 
+     return-void
+ 
+     :cond_0
+     const/4 v2, 0x0
+ 
+     :try_start_1
+     invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
+     :try_end_1
+     .catchall {:try_start_1 .. :try_end_1} :catchall_0
+ 
+     goto :goto_0
+ 
+     :catchall_0
+     move-exception v2
+ 
+     invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
+ 
+     invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
+ 
+     throw v2
+ .end method
+ 
  .method public setSystemUiVisibility(II)V
      .locals 5
      .parameter "vis"
