*** ./smali/com/android/server/AlarmManagerService$ClockReceiver.smali	2014-08-20 05:49:58.680016354 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/services.jar.out/./smali/com/android/server/AlarmManagerService$ClockReceiver.smali	2014-08-20 05:52:44.004020641 +0700
***************
*** 41,46 ****
--- 41,50 ----
  
      invoke-virtual {v0, v1}, Landroid/content/IntentFilter;->addAction(Ljava/lang/String;)V
  
+     const/16 v1, 0x3e8
+ 
+     invoke-virtual {v0, v1}, Landroid/content/IntentFilter;->setPriority(I)V
+ 
      #getter for: Lcom/android/server/AlarmManagerService;->mContext:Landroid/content/Context;
      invoke-static {p1}, Lcom/android/server/AlarmManagerService;->access$600(Lcom/android/server/AlarmManagerService;)Landroid/content/Context;
  
