*** ./smali/com/android/server/ConnectivityService.smali	2014-08-20 05:49:57.652016327 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/services.jar.out/./smali/com/android/server/ConnectivityService.smali	2014-08-20 05:52:42.988020614 +0700
***************
*** 593,598 ****
--- 593,600 ----
  
      iput-object v2, v0, Lcom/android/server/ConnectivityService;->mUserIntentReceiver:Landroid/content/BroadcastReceiver;
  
+     invoke-static/range {p1 .. p1}, Lcom/android/server/Injector$ConnectivityServiceHook;->before_ConnectivityService(Landroid/content/Context;)V
+ 
      const-string v2, "ConnectivityService starting up"
  
      invoke-static {v2}, Lcom/android/server/ConnectivityService;->log(Ljava/lang/String;)V
***************
*** 3248,3254 ****
  
      .end local v1           #tracker:Landroid/net/NetworkStateTracker;
      :cond_0
!     return-object v0
  .end method
  
  .method private getNetworkStateUnchecked(I)Landroid/net/NetworkState;
--- 3250,3260 ----
  
      .end local v1           #tracker:Landroid/net/NetworkStateTracker;
      :cond_0
!     invoke-static {p0, p2, v0}, Lcom/android/server/Injector$ConnectivityServiceHook;->transferNetworkInfo(Lcom/android/server/ConnectivityService;ILandroid/net/NetworkInfo;)Landroid/net/NetworkInfo;
! 
!     move-result-object v2
! 
!     return-object v2
  .end method
  
  .method private getNetworkStateUnchecked(I)Landroid/net/NetworkState;
***************
*** 9664,9669 ****
--- 9670,9681 ----
      move-result v10
  
      .local v10, usedNetworkType:I
+     iget v12, p1, Lcom/android/server/ConnectivityService$FeatureUser;->mUid:I
+ 
+     iget v13, p1, Lcom/android/server/ConnectivityService$FeatureUser;->mPid:I
+ 
+     invoke-static {v12, v13, v10}, Lcom/android/server/Injector$ConnectivityServiceHook;->stopUsingNetworkFeature(III)V
+ 
      iget-object v12, p0, Lcom/android/server/ConnectivityService;->mNetTrackers:[Landroid/net/NetworkStateTracker;
  
      aget-object v8, v12, v10
***************
*** 12260,12265 ****
--- 12272,12286 ----
      throw v7
  .end method
  
+ .method getContext()Landroid/content/Context;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/server/ConnectivityService;->mContext:Landroid/content/Context;
+ 
+     return-object v0
+ .end method
+ 
  .method public getGlobalProxy()Landroid/net/ProxyProperties;
      .locals 2
  
***************
*** 12623,12628 ****
--- 12644,12658 ----
      return-object v0
  .end method
  
+ .method getNetTrackers()[Landroid/net/NetworkStateTracker;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/server/ConnectivityService;->mNetTrackers:[Landroid/net/NetworkStateTracker;
+ 
+     return-object v0
+ .end method
+ 
  .method public getNetworkInfo(I)Landroid/net/NetworkInfo;
      .locals 2
      .parameter "networkType"
***************
*** 15110,15118 ****
      :try_end_8
      .catchall {:try_start_8 .. :try_end_8} :catchall_2
  
      if-ltz v13, :cond_d
  
-     :try_start_9
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/server/ConnectivityService;->mHandler:Lcom/android/server/ConnectivityService$InternalHandler;
--- 15140,15150 ----
      :try_end_8
      .catchall {:try_start_8 .. :try_end_8} :catchall_2
  
+     :try_start_9
+     invoke-static/range {v20 .. v20}, Lcom/android/server/Injector$ConnectivityServiceHook;->startUsingNetworkFeature(I)V
+ 
      if-ltz v13, :cond_d
  
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/server/ConnectivityService;->mHandler:Lcom/android/server/ConnectivityService$InternalHandler;
