*** ./smali/com/android/server/LightsService$Light.smali	2014-08-20 05:49:57.956016335 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/services.jar.out/./smali/com/android/server/LightsService$Light.smali	2014-08-20 05:52:43.284020622 +0700
***************
*** 1,4 ****
! .class public final Lcom/android/server/LightsService$Light;
  .super Ljava/lang/Object;
  .source "LightsService.java"
  
--- 1,4 ----
! .class public Lcom/android/server/LightsService$Light;
  .super Ljava/lang/Object;
  .source "LightsService.java"
  
***************
*** 9,15 ****
  .end annotation
  
  .annotation system Ldalvik/annotation/InnerClass;
!     accessFlags = 0x11
      name = "Light"
  .end annotation
  
--- 9,15 ----
  .end annotation
  
  .annotation system Ldalvik/annotation/InnerClass;
!     accessFlags = 0x1
      name = "Light"
  .end annotation
  
***************
*** 46,51 ****
--- 46,63 ----
      return-void
  .end method
  
+ .method constructor <init>(Lcom/android/server/LightsService;II)V
+     .locals 0
+     .parameter
+     .parameter "id"
+     .parameter "unused"
+ 
+     .prologue
+     invoke-direct {p0, p1, p2}, Lcom/android/server/LightsService$Light;-><init>(Lcom/android/server/LightsService;I)V
+ 
+     return-void
+ .end method
+ 
  .method synthetic constructor <init>(Lcom/android/server/LightsService;ILcom/android/server/LightsService$1;)V
      .locals 0
      .parameter "x0"
***************
*** 68,74 ****
      return-void
  .end method
  
! .method private setLightLocked(IIIII)V
      .locals 7
      .parameter "color"
      .parameter "mode"
--- 80,86 ----
      return-void
  .end method
  
! .method setLightLocked(IIIII)V
      .locals 7
      .parameter "color"
      .parameter "mode"
***************
*** 147,153 ****
  
      move-object v0, p0
  
!     invoke-direct/range {v0 .. v5}, Lcom/android/server/LightsService$Light;->setLightLocked(IIIII)V
  
      monitor-exit p0
  
--- 159,165 ----
  
      move-object v0, p0
  
!     invoke-virtual/range {v0 .. v5}, Lcom/android/server/LightsService$Light;->setLightLocked(IIIII)V
  
      monitor-exit p0
  
***************
*** 207,213 ****
  
      move v3, p2
  
!     invoke-direct/range {v0 .. v5}, Lcom/android/server/LightsService$Light;->setLightLocked(IIIII)V
  
      iget-object v0, p0, Lcom/android/server/LightsService$Light;->this$0:Lcom/android/server/LightsService;
  
--- 219,225 ----
  
      move v3, p2
  
!     invoke-virtual/range {v0 .. v5}, Lcom/android/server/LightsService$Light;->setLightLocked(IIIII)V
  
      iget-object v0, p0, Lcom/android/server/LightsService$Light;->this$0:Lcom/android/server/LightsService;
  
***************
*** 294,300 ****
      move v5, p2
  
      :try_start_0
!     invoke-direct/range {v0 .. v5}, Lcom/android/server/LightsService$Light;->setLightLocked(IIIII)V
  
      monitor-exit p0
  
--- 306,312 ----
      move v5, p2
  
      :try_start_0
!     invoke-virtual/range {v0 .. v5}, Lcom/android/server/LightsService$Light;->setLightLocked(IIIII)V
  
      monitor-exit p0
  
***************
*** 330,336 ****
      move v1, p1
  
      :try_start_0
!     invoke-direct/range {v0 .. v5}, Lcom/android/server/LightsService$Light;->setLightLocked(IIIII)V
  
      monitor-exit p0
  
--- 342,348 ----
      move v1, p1
  
      :try_start_0
!     invoke-virtual/range {v0 .. v5}, Lcom/android/server/LightsService$Light;->setLightLocked(IIIII)V
  
      monitor-exit p0
  
***************
*** 369,375 ****
      move v4, p4
  
      :try_start_0
!     invoke-direct/range {v0 .. v5}, Lcom/android/server/LightsService$Light;->setLightLocked(IIIII)V
  
      monitor-exit p0
  
--- 381,387 ----
      move v4, p4
  
      :try_start_0
!     invoke-virtual/range {v0 .. v5}, Lcom/android/server/LightsService$Light;->setLightLocked(IIIII)V
  
      monitor-exit p0
  
***************
*** 404,410 ****
      move-object v0, p0
  
      :try_start_0
!     invoke-direct/range {v0 .. v5}, Lcom/android/server/LightsService$Light;->setLightLocked(IIIII)V
  
      monitor-exit p0
  
--- 416,422 ----
      move-object v0, p0
  
      :try_start_0
!     invoke-virtual/range {v0 .. v5}, Lcom/android/server/LightsService$Light;->setLightLocked(IIIII)V
  
      monitor-exit p0
  
