*** ./smali/com/android/server/NotificationManagerService.smali	2014-08-20 05:49:36.012015766 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/services.jar.out/./smali/com/android/server/NotificationManagerService.smali	2014-08-20 05:52:21.308020052 +0700
***************
*** 2932,2940 ****
      goto/16 :goto_0
  .end method
  
! .method private scheduleTimeoutLocked(Lcom/android/server/NotificationManagerService$ToastRecord;)V
      .locals 5
      .parameter "r"
  
      .prologue
      iget-object v3, p0, Lcom/android/server/NotificationManagerService;->mHandler:Lcom/android/server/NotificationManagerService$WorkerHandler;
--- 2932,2941 ----
      goto/16 :goto_0
  .end method
  
! .method private scheduleTimeoutLocked(Lcom/android/server/NotificationManagerService$ToastRecord;Z)V
      .locals 5
      .parameter "r"
+     .parameter "immediate"
  
      .prologue
      iget-object v3, p0, Lcom/android/server/NotificationManagerService;->mHandler:Lcom/android/server/NotificationManagerService$WorkerHandler;
***************
*** 2950,2962 ****
      move-result-object v2
  
      .local v2, m:Landroid/os/Message;
!     iget v3, p1, Lcom/android/server/NotificationManagerService$ToastRecord;->duration:I
! 
!     const/4 v4, 0x1
  
!     if-ne v3, v4, :cond_0
! 
!     const-wide/16 v0, 0xdac
  
      .local v0, delay:J
      :goto_0
--- 2951,2959 ----
      move-result-object v2
  
      .local v2, m:Landroid/os/Message;
!     if-eqz p2, :cond_0
  
!     const-wide/16 v0, 0x0
  
      .local v0, delay:J
      :goto_0
***************
*** 2968,2976 ****
  
      .end local v0           #delay:J
      :cond_0
!     const-wide/16 v0, 0x7d0
  
      goto :goto_0
  .end method
  
  .method private sendAccessibilityEvent(Landroid/app/Notification;Ljava/lang/CharSequence;)V
--- 2965,2987 ----
  
      .end local v0           #delay:J
      :cond_0
!     iget v3, p1, Lcom/android/server/NotificationManagerService$ToastRecord;->duration:I
! 
!     const/4 v4, 0x1
! 
!     if-ne v3, v4, :cond_1
! 
!     const/16 v3, 0xdac
! 
!     :goto_1
!     int-to-long v0, v3
  
      goto :goto_0
+ 
+     :cond_1
+     const/16 v3, 0x7d0
+ 
+     goto :goto_1
  .end method
  
  .method private sendAccessibilityEvent(Landroid/app/Notification;Ljava/lang/CharSequence;)V
***************
*** 3059,3065 ****
  
      invoke-interface {v3}, Landroid/app/ITransientNotification;->show()V
  
!     invoke-direct {p0, v2}, Lcom/android/server/NotificationManagerService;->scheduleTimeoutLocked(Lcom/android/server/NotificationManagerService$ToastRecord;)V
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
--- 3070,3078 ----
  
      invoke-interface {v3}, Landroid/app/ITransientNotification;->show()V
  
!     const/4 v3, 0x0
! 
!     invoke-direct {p0, v2, v3}, Lcom/android/server/NotificationManagerService;->scheduleTimeoutLocked(Lcom/android/server/NotificationManagerService$ToastRecord;Z)V
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
***************
*** 3369,3374 ****
--- 3382,3395 ----
  
      iget v1, p0, Lcom/android/server/NotificationManagerService;->mDefaultNotificationLedOff:I
  
+     invoke-static {p0, v3}, Lcom/android/server/Injector$NotificationManagerServiceHook;->updateNotificationLight(Lcom/android/server/NotificationManagerService;Landroid/app/Notification;)V
+ 
+     iget v0, v3, Landroid/app/Notification;->ledARGB:I
+ 
+     iget v2, v3, Landroid/app/Notification;->ledOnMS:I
+ 
+     iget v1, v3, Landroid/app/Notification;->ledOffMS:I
+ 
      :cond_4
      iget-boolean v5, p0, Lcom/android/server/NotificationManagerService;->mNotificationPulseEnabled:Z
  
***************
*** 3410,3415 ****
--- 3431,3463 ----
  
  
  # virtual methods
+ .method public areNotificationsEnabled(Ljava/lang/String;)Z
+     .locals 1
+     .parameter "pkg"
+ 
+     .prologue
+     const/4 v0, 0x0
+ 
+     invoke-virtual {p0, p1, v0}, Lcom/android/server/NotificationManagerService;->areNotificationsEnabledForPackage(Ljava/lang/String;I)Z
+ 
+     move-result v0
+ 
+     return v0
+ .end method
+ 
+ .method public areNotificationsEnabled(Ljava/lang/String;I)Z
+     .locals 1
+     .parameter "pkg"
+     .parameter "uid"
+ 
+     .prologue
+     invoke-virtual {p0, p1, p2}, Lcom/android/server/NotificationManagerService;->areNotificationsEnabledForPackage(Ljava/lang/String;I)Z
+ 
+     move-result v0
+ 
+     return v0
+ .end method
+ 
  .method public areNotificationsEnabledForPackage(Ljava/lang/String;I)Z
      .locals 2
      .parameter "pkg"
***************
*** 3792,3797 ****
--- 3840,3870 ----
      goto :goto_2
  .end method
  
+ .method cancelCurrentToast(I)V
+     .locals 2
+     .parameter "index"
+ 
+     .prologue
+     if-eqz p1, :cond_0
+ 
+     iget-object v0, p0, Lcom/android/server/NotificationManagerService;->mToastQueue:Ljava/util/ArrayList;
+ 
+     const/4 v1, 0x0
+ 
+     invoke-virtual {v0, v1}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
+ 
+     move-result-object v0
+ 
+     check-cast v0, Lcom/android/server/NotificationManagerService$ToastRecord;
+ 
+     const/4 v1, 0x1
+ 
+     invoke-direct {p0, v0, v1}, Lcom/android/server/NotificationManagerService;->scheduleTimeoutLocked(Lcom/android/server/NotificationManagerService$ToastRecord;Z)V
+ 
+     :cond_0
+     return-void
+ .end method
+ 
  .method public cancelNotificationFromListener(Landroid/service/notification/INotificationListener;Ljava/lang/String;Ljava/lang/String;I)V
      .locals 11
      .parameter "token"
***************
*** 5945,5950 ****
--- 6018,6027 ----
      move-object/from16 v0, p0
  
      invoke-direct {v0, v6}, Lcom/android/server/NotificationManagerService;->keepProcessAliveLocked(I)V
+ 
+     move-object/from16 v0, p0
+ 
+     invoke-static {v0, v9}, Lcom/android/server/Injector$NotificationManagerServiceHook;->cancelCurrentToast(Lcom/android/server/NotificationManagerService;I)V
      :try_end_5
      .catchall {:try_start_5 .. :try_end_5} :catchall_1
  
***************
*** 6143,6148 ****
--- 6220,6234 ----
      throw v6
  .end method
  
+ .method getDefaultNotificationColor()I
+     .locals 1
+ 
+     .prologue
+     iget v0, p0, Lcom/android/server/NotificationManagerService;->mDefaultNotificationColor:I
+ 
+     return v0
+ .end method
+ 
  .method public getHistoricalNotifications(Ljava/lang/String;I)[Landroid/service/notification/StatusBarNotification;
      .locals 5
      .parameter "callingPkg"
