*** ./smali/com/android/server/StatusBarManagerService.smali	2014-08-20 05:49:54.516016246 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/services.jar.out/./smali/com/android/server/StatusBarManagerService.smali	2014-08-20 05:52:39.872020533 +0700
***************
*** 1890,1895 ****
--- 1890,1908 ----
      return-void
  .end method
  
+ .method public setStatus(ILandroid/os/IBinder;Ljava/lang/String;Landroid/os/Bundle;)V
+     .locals 0
+     .parameter "what"
+     .parameter "token"
+     .parameter "action"
+     .parameter "ext"
+ 
+     .prologue
+     invoke-static {p0, p1, p2, p3, p4}, Lcom/android/server/Injector$StatusBarManagerServiceHook;->setStatus(Lcom/android/server/StatusBarManagerService;ILandroid/os/IBinder;Ljava/lang/String;Landroid/os/Bundle;)V
+ 
+     return-void
+ .end method
+ 
  .method public setSystemUiVisibility(II)V
      .locals 5
      .parameter "vis"
