*** ./smali/com/android/server/UiModeManagerService.smali	2014-08-20 05:49:52.844016202 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/services.jar.out/./smali/com/android/server/UiModeManagerService.smali	2014-08-20 05:52:38.180020489 +0700
***************
*** 30,36 ****
  
  .field private mCurUiMode:I
  
! .field private final mDefaultUiModeType:I
  
  .field private final mDeskModeKeepsScreenOn:Z
  
--- 30,36 ----
  
  .field private mCurUiMode:I
  
! .field private mDefaultUiModeType:I
  
  .field private final mDeskModeKeepsScreenOn:Z
  
***************
*** 291,296 ****
--- 291,298 ----
  
      invoke-virtual {v0, v1, v2}, Lcom/android/server/TwilightService;->registerListener(Lcom/android/server/TwilightService$TwilightListener;Landroid/os/Handler;)V
  
+     invoke-static {p0, p1}, Lcom/android/server/Injector$UiModeManagerServiceHook;->registerUIModeScaleChangedOjbserver(Lcom/android/server/UiModeManagerService;Landroid/content/Context;)V
+ 
      return-void
  
      :cond_0
***************
*** 1728,1733 ****
--- 1730,1745 ----
      throw v0
  .end method
  
+ .method setDefaultUiModeType(I)V
+     .locals 0
+     .parameter "type"
+ 
+     .prologue
+     iput p1, p0, Lcom/android/server/UiModeManagerService;->mDefaultUiModeType:I
+ 
+     return-void
+ .end method
+ 
  .method public setNightMode(I)V
      .locals 5
      .parameter "mode"
