*** ./smali/com/android/server/location/GpsLocationProvider.smali	2014-08-20 05:49:47.020016051 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/services.jar.out/./smali/com/android/server/location/GpsLocationProvider.smali	2014-08-20 05:52:32.428020340 +0700
***************
*** 3979,3984 ****
--- 3979,3986 ----
  
      invoke-virtual {v2, v6, v8}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Z)Landroid/content/Intent;
  
+     invoke-static {p0, v2}, Lcom/android/server/location/Injector$GpsLocationProviderHook;->appendUidExtra(Lcom/android/server/location/GpsLocationProvider;Landroid/content/Intent;)V
+ 
      iget-object v6, p0, Lcom/android/server/location/GpsLocationProvider;->mContext:Landroid/content/Context;
  
      sget-object v8, Landroid/os/UserHandle;->ALL:Landroid/os/UserHandle;
***************
*** 5590,5595 ****
--- 5592,5606 ----
      throw v0
  .end method
  
+ .method getClientSource()Landroid/os/WorkSource;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/server/location/GpsLocationProvider;->mClientSource:Landroid/os/WorkSource;
+ 
+     return-object v0
+ .end method
+ 
  .method public getGpsGeofenceProxy()Landroid/location/IGpsGeofenceHardware;
      .locals 1
  
***************
*** 5617,5622 ****
--- 5628,5642 ----
      return-object v0
  .end method
  
+ .method getNavigating()Z
+     .locals 1
+ 
+     .prologue
+     iget-boolean v0, p0, Lcom/android/server/location/GpsLocationProvider;->mNavigating:Z
+ 
+     return v0
+ .end method
+ 
  .method public getNetInitiatedListener()Landroid/location/INetInitiatedListener;
      .locals 1
  
