*** ./smali/android/net/wifi/WifiConfiguration.smali	2014-08-20 05:48:40.108014316 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/framework.jar.out/./smali/android/net/wifi/WifiConfiguration.smali	2014-08-20 05:51:25.200018597 +0700
***************
*** 47,52 ****
--- 47,56 ----
  
  .field private static final TAG:Ljava/lang/String; = "WifiConfiguration"
  
+ .field public static final WAPI_ASCII_PASSWORD:I = 0x0
+ 
+ .field public static final WAPI_HEX_PASSWORD:I = 0x1
+ 
  .field public static final bssidVarName:Ljava/lang/String; = "bssid"
  
  .field public static final hiddenSSIDVarName:Ljava/lang/String; = "scan_ssid"
***************
*** 57,62 ****
--- 61,76 ----
  
  .field public static final ssidVarName:Ljava/lang/String; = "ssid"
  
+ .field public static final wapiAsCertVarName:Ljava/lang/String; = "as_cert_file"
+ 
+ .field public static final wapiCertIndexVarName:Ljava/lang/String; = "cert_index"
+ 
+ .field public static final wapiPskTypeVarName:Ljava/lang/String; = "wapi_key_type"
+ 
+ .field public static final wapiPskVarName:Ljava/lang/String; = "wapi_psk"
+ 
+ .field public static final wapiUserCertVarName:Ljava/lang/String; = "user_cert_file"
+ 
  .field public static final wepKeyVarNames:[Ljava/lang/String; = null
  
  .field public static final wepTxKeyIdxVarName:Ljava/lang/String; = "wep_tx_keyidx"
***************
*** 97,102 ****
--- 111,126 ----
  
  .field public status:I
  
+ .field public wapiASCert:Ljava/lang/String;
+ 
+ .field public wapiCertIndex:I
+ 
+ .field public wapiPsk:Ljava/lang/String;
+ 
+ .field public wapiPskType:I
+ 
+ .field public wapiUserCert:Ljava/lang/String;
+ 
  .field public wepKeys:[Ljava/lang/String;
  
  .field public wepTxKeyIndex:I
***************
*** 225,230 ****
--- 249,264 ----
      goto :goto_0
  
      :cond_0
+     const/4 v1, -0x1
+ 
+     iput-object v2, p0, Landroid/net/wifi/WifiConfiguration;->wapiASCert:Ljava/lang/String;
+ 
+     iput-object v2, p0, Landroid/net/wifi/WifiConfiguration;->wapiUserCert:Ljava/lang/String;
+ 
+     iput v1, p0, Landroid/net/wifi/WifiConfiguration;->wapiCertIndex:I
+ 
+     iput v1, p0, Landroid/net/wifi/WifiConfiguration;->wapiPskType:I
+ 
      new-instance v1, Landroid/net/wifi/WifiEnterpriseConfig;
  
      invoke-direct {v1}, Landroid/net/wifi/WifiEnterpriseConfig;-><init>()V
***************
*** 281,286 ****
--- 315,332 ----
  
      iput-object v1, p0, Landroid/net/wifi/WifiConfiguration;->preSharedKey:Ljava/lang/String;
  
+     const/4 v0, 0x0
+ 
+     const/4 v1, -0x1
+ 
+     iput-object v0, p0, Landroid/net/wifi/WifiConfiguration;->wapiASCert:Ljava/lang/String;
+ 
+     iput-object v0, p0, Landroid/net/wifi/WifiConfiguration;->wapiUserCert:Ljava/lang/String;
+ 
+     iput v1, p0, Landroid/net/wifi/WifiConfiguration;->wapiCertIndex:I
+ 
+     iput v1, p0, Landroid/net/wifi/WifiConfiguration;->wapiPskType:I
+ 
      const/4 v1, 0x4
  
      new-array v1, v1, [Ljava/lang/String;
***************
*** 595,600 ****
--- 641,676 ----
      goto :goto_0
  
      :cond_4
+     iget-object v0, p0, Landroid/net/wifi/WifiConfiguration;->allowedKeyManagement:Ljava/util/BitSet;
+ 
+     const/4 v1, 0x5
+ 
+     invoke-virtual {v0, v1}, Ljava/util/BitSet;->get(I)Z
+ 
+     move-result v0
+ 
+     if-eqz v0, :cond_miui_0
+ 
+     const/4 v0, 0x5
+ 
+     goto :goto_0
+ 
+     :cond_miui_0
+     iget-object v0, p0, Landroid/net/wifi/WifiConfiguration;->allowedKeyManagement:Ljava/util/BitSet;
+ 
+     const/4 v1, 0x6
+ 
+     invoke-virtual {v0, v1}, Ljava/util/BitSet;->get(I)Z
+ 
+     move-result v0
+ 
+     if-eqz v0, :cond_miui_1
+ 
+     const/4 v0, 0x6
+ 
+     goto :goto_0
+ 
+     :cond_miui_1
      const/4 v0, 0x0
  
      goto :goto_0
***************
*** 1466,1471 ****
--- 1542,1563 ----
  
      invoke-virtual {p1, v4}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     iget-object v4, p0, Landroid/net/wifi/WifiConfiguration;->wapiASCert:Ljava/lang/String;
+ 
+     invoke-virtual {p1, v4}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
+ 
+     iget-object v4, p0, Landroid/net/wifi/WifiConfiguration;->wapiUserCert:Ljava/lang/String;
+ 
+     invoke-virtual {p1, v4}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
+ 
+     iget-object v4, p0, Landroid/net/wifi/WifiConfiguration;->wapiPsk:Ljava/lang/String;
+ 
+     invoke-virtual {p1, v4}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
+ 
+     iget v4, p0, Landroid/net/wifi/WifiConfiguration;->wapiPskType:I
+ 
+     invoke-virtual {p1, v4}, Landroid/os/Parcel;->writeInt(I)V
+ 
      iget-object v0, p0, Landroid/net/wifi/WifiConfiguration;->wepKeys:[Ljava/lang/String;
  
      .local v0, arr$:[Ljava/lang/String;
