*** ./smali/android/net/wifi/WifiMonitor$MonitorThread.smali	2014-08-20 05:48:40.220014319 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/framework.jar.out/./smali/android/net/wifi/WifiMonitor$MonitorThread.smali	2014-08-20 05:51:25.308018600 +0700
***************
*** 100,105 ****
--- 100,129 ----
      return v5
  
      :cond_1
+     const-string v6, "WAPI:"
+ 
+     invoke-virtual {p1, v6}, Ljava/lang/String;->startsWith(Ljava/lang/String;)Z
+ 
+     move-result v6
+ 
+     if-eqz v6, :cond_miui
+ 
+     const-string v6, "pre-shared key may be incorrect"
+ 
+     invoke-virtual {p1, v6}, Ljava/lang/String;->indexOf(Ljava/lang/String;)I
+ 
+     move-result v6
+ 
+     if-lez v6, :cond_miui
+ 
+     iget-object v6, p0, Landroid/net/wifi/WifiMonitor$MonitorThread;->mStateMachine:Lcom/android/internal/util/StateMachine;
+ 
+     invoke-virtual {v6, v9}, Lcom/android/internal/util/StateMachine;->sendMessage(I)V
+ 
+     goto :goto_0
+ 
+     :cond_miui
+ 
      const-string v6, "WPS-SUCCESS"
  
      invoke-virtual {p1, v6}, Ljava/lang/String;->startsWith(Ljava/lang/String;)Z
