*** ./smali/android/accounts/ChooseTypeAndAccountActivity.smali	2014-08-20 05:48:47.024014495 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/framework.jar.out/./smali/android/accounts/ChooseTypeAndAccountActivity.smali	2014-08-20 05:51:32.180018778 +0700
***************
*** 825,830 ****
--- 825,832 ----
      invoke-direct {v0, p0, v1}, Landroid/content/Intent;-><init>(Landroid/content/Context;Ljava/lang/Class;)V
  
      .local v0, intent:Landroid/content/Intent;
+     invoke-static {v0}, Landroid/accounts/Injector$ChooseTypeAndAccountActivityHook;->useMiuiChooseAccountTypeActivity(Landroid/content/Intent;)V
+ 
      const/high16 v1, 0x8
  
      invoke-virtual {v0, v1}, Landroid/content/Intent;->setFlags(I)Landroid/content/Intent;
