*** ./smali/android/text/util/Linkify$4.smali	2014-08-20 05:48:57.104014757 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/framework.jar.out/./smali/android/text/util/Linkify$4.smali	2014-08-20 05:51:42.208019038 +0700
***************
*** 21,27 ****
          "Ljava/lang/Object;",
          "Ljava/util/Comparator",
          "<",
!         "Landroid/text/util/LinkSpec;",
          ">;"
      }
  .end annotation
--- 21,27 ----
          "Ljava/lang/Object;",
          "Ljava/util/Comparator",
          "<",
!         "Landroid/text/util/Linkify$LinkSpec;",
          ">;"
      }
  .end annotation
***************
*** 39,45 ****
  
  
  # virtual methods
! .method public final compare(Landroid/text/util/LinkSpec;Landroid/text/util/LinkSpec;)I
      .locals 4
      .parameter "a"
      .parameter "b"
--- 39,45 ----
  
  
  # virtual methods
! .method public final compare(Landroid/text/util/Linkify$LinkSpec;Landroid/text/util/Linkify$LinkSpec;)I
      .locals 4
      .parameter "a"
      .parameter "b"
***************
*** 49,57 ****
  
      const/4 v0, -0x1
  
!     iget v2, p1, Landroid/text/util/LinkSpec;->start:I
  
!     iget v3, p2, Landroid/text/util/LinkSpec;->start:I
  
      if-ge v2, v3, :cond_1
  
--- 49,57 ----
  
      const/4 v0, -0x1
  
!     iget v2, p1, Landroid/text/util/Linkify$LinkSpec;->start:I
  
!     iget v3, p2, Landroid/text/util/Linkify$LinkSpec;->start:I
  
      if-ge v2, v3, :cond_1
  
***************
*** 60,68 ****
      return v0
  
      :cond_1
!     iget v2, p1, Landroid/text/util/LinkSpec;->start:I
  
!     iget v3, p2, Landroid/text/util/LinkSpec;->start:I
  
      if-le v2, v3, :cond_2
  
--- 60,68 ----
      return v0
  
      :cond_1
!     iget v2, p1, Landroid/text/util/Linkify$LinkSpec;->start:I
  
!     iget v3, p2, Landroid/text/util/Linkify$LinkSpec;->start:I
  
      if-le v2, v3, :cond_2
  
***************
*** 71,79 ****
      goto :goto_0
  
      :cond_2
!     iget v2, p1, Landroid/text/util/LinkSpec;->end:I
  
!     iget v3, p2, Landroid/text/util/LinkSpec;->end:I
  
      if-ge v2, v3, :cond_3
  
--- 71,79 ----
      goto :goto_0
  
      :cond_2
!     iget v2, p1, Landroid/text/util/Linkify$LinkSpec;->end:I
  
!     iget v3, p2, Landroid/text/util/Linkify$LinkSpec;->end:I
  
      if-ge v2, v3, :cond_3
  
***************
*** 82,90 ****
      goto :goto_0
  
      :cond_3
!     iget v1, p1, Landroid/text/util/LinkSpec;->end:I
  
!     iget v2, p2, Landroid/text/util/LinkSpec;->end:I
  
      if-gt v1, v2, :cond_0
  
--- 82,90 ----
      goto :goto_0
  
      :cond_3
!     iget v1, p1, Landroid/text/util/Linkify$LinkSpec;->end:I
  
!     iget v2, p2, Landroid/text/util/Linkify$LinkSpec;->end:I
  
      if-gt v1, v2, :cond_0
  
***************
*** 99,111 ****
      .parameter "x1"
  
      .prologue
!     check-cast p1, Landroid/text/util/LinkSpec;
  
      .end local p1
!     check-cast p2, Landroid/text/util/LinkSpec;
  
      .end local p2
!     invoke-virtual {p0, p1, p2}, Landroid/text/util/Linkify$4;->compare(Landroid/text/util/LinkSpec;Landroid/text/util/LinkSpec;)I
  
      move-result v0
  
--- 99,111 ----
      .parameter "x1"
  
      .prologue
!     check-cast p1, Landroid/text/util/Linkify$LinkSpec;
  
      .end local p1
!     check-cast p2, Landroid/text/util/Linkify$LinkSpec;
  
      .end local p2
!     invoke-virtual {p0, p1, p2}, Landroid/text/util/Linkify$4;->compare(Landroid/text/util/Linkify$LinkSpec;Landroid/text/util/Linkify$LinkSpec;)I
  
      move-result v0
  
