*** ./smali/com/android/internal/widget/multiwaveview/GlowPadView.smali	2014-08-20 05:49:26.644015523 +0700
--- /home/fantastical/workspace/patchrom/taoshan/temp/new_smali/framework2.jar.out/./smali/com/android/internal/widget/multiwaveview/GlowPadView.smali	2014-08-20 05:52:11.824019806 +0700
***************
*** 662,668 ****
      .parameter "x3"
  
      .prologue
!     invoke-direct {p0, p1, p2, p3}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->switchToState(IFF)V
  
      return-void
  .end method
--- 662,668 ----
      .parameter "x3"
  
      .prologue
!     invoke-virtual {p0, p1, p2, p3}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->switchToState(IFF)V
  
      return-void
  .end method
***************
*** 716,722 ****
      .parameter "x2"
  
      .prologue
!     invoke-direct {p0, p1, p2}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->hideTargets(ZZ)V
  
      return-void
  .end method
--- 716,722 ----
      .parameter "x2"
  
      .prologue
!     invoke-virtual {p0, p1, p2}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->hideTargets(ZZ)V
  
      return-void
  .end method
***************
*** 1193,1199 ****
      return v0
  .end method
  
! .method private doFinish()V
      .locals 7
  
      .prologue
--- 1193,1199 ----
      return v0
  .end method
  
! .method doFinish()V
      .locals 7
  
      .prologue
***************
*** 1254,1260 ****
  
      invoke-direct {p0, v6, v3, v5, v4}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->hideGlow(IIFLandroid/animation/Animator$AnimatorListener;)V
  
!     invoke-direct {p0, v2, v3}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->hideTargets(ZZ)V
  
      goto :goto_1
  .end method
--- 1254,1260 ----
  
      invoke-direct {p0, v6, v3, v5, v4}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->hideGlow(IIFLandroid/animation/Animator$AnimatorListener;)V
  
!     invoke-virtual {p0, v2, v3}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->hideTargets(ZZ)V
  
      goto :goto_1
  .end method
***************
*** 1749,1755 ****
  
      move-result v3
  
!     invoke-direct {p0, v1, v2, v3}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->switchToState(IFF)V
  
      return-void
  .end method
--- 1749,1755 ----
  
      move-result v3
  
!     invoke-virtual {p0, v1, v2, v3}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->switchToState(IFF)V
  
      return-void
  .end method
***************
*** 1776,1782 ****
      .local v2, eventY:F
      const/4 v3, 0x1
  
!     invoke-direct {p0, v3, v1, v2}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->switchToState(IFF)V
  
      invoke-direct {p0, v1, v2}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->trySwitchToFirstTouchState(FF)Z
  
--- 1776,1782 ----
      .local v2, eventY:F
      const/4 v3, 0x1
  
!     invoke-virtual {p0, v3, v1, v2}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->switchToState(IFF)V
  
      invoke-direct {p0, v1, v2}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->trySwitchToFirstTouchState(FF)Z
  
***************
*** 1803,1809 ****
      goto :goto_0
  .end method
  
! .method private handleMove(Landroid/view/MotionEvent;)V
      .locals 39
      .parameter "event"
  
--- 1803,1809 ----
      goto :goto_0
  .end method
  
! .method handleMove(Landroid/view/MotionEvent;)V
      .locals 39
      .parameter "event"
  
***************
*** 2321,2327 ****
  
      move/from16 v3, v32
  
!     invoke-direct {v0, v1, v2, v3}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->switchToState(IFF)V
  
      move-object/from16 v0, p0
  
--- 2321,2327 ----
  
      move/from16 v3, v32
  
!     invoke-virtual {v0, v1, v2, v3}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->switchToState(IFF)V
  
      move-object/from16 v0, p0
  
***************
*** 2494,2500 ****
  
      move/from16 v2, v34
  
!     invoke-direct {v0, v6, v1, v2, v5}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->updateTargetPosition(IFFF)V
  
      :cond_10
      move-object/from16 v0, p0
--- 2494,2500 ----
  
      move/from16 v2, v34
  
!     invoke-virtual {v0, v6, v1, v2, v5}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->updateTargetPosition(IFFF)V
  
      :cond_10
      move-object/from16 v0, p0
***************
*** 2546,2552 ****
  
      move/from16 v3, v32
  
!     invoke-direct {v0, v1, v2, v3}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->switchToState(IFF)V
  
      move-object/from16 v0, p0
  
--- 2546,2552 ----
  
      move/from16 v3, v32
  
!     invoke-virtual {v0, v1, v2, v3}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->switchToState(IFF)V
  
      move-object/from16 v0, p0
  
***************
*** 2587,2593 ****
  
      move-result v3
  
!     invoke-direct {p0, v1, v2, v3}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->switchToState(IFF)V
  
      :cond_0
      return-void
--- 2587,2593 ----
  
      move-result v3
  
!     invoke-virtual {p0, v1, v2, v3}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->switchToState(IFF)V
  
      :cond_0
      return-void
***************
*** 2722,2728 ****
      return-void
  .end method
  
! .method private hideTargets(ZZ)V
      .locals 17
      .parameter "animate"
      .parameter "expanded"
--- 2722,2728 ----
      return-void
  .end method
  
! .method hideTargets(ZZ)V
      .locals 17
      .parameter "animate"
      .parameter "expanded"
***************
*** 3630,3636 ****
      return-void
  .end method
  
! .method private showTargets(Z)V
      .locals 13
      .parameter "animate"
  
--- 3630,3636 ----
      return-void
  .end method
  
! .method showTargets(Z)V
      .locals 13
      .parameter "animate"
  
***************
*** 4183,4189 ****
      return-void
  .end method
  
! .method private switchToState(IFF)V
      .locals 6
      .parameter "state"
      .parameter "x"
--- 4183,4189 ----
      return-void
  .end method
  
! .method switchToState(IFF)V
      .locals 6
      .parameter "state"
      .parameter "x"
***************
*** 4237,4243 ****
  
      invoke-direct {p0}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->deactivateTargets()V
  
!     invoke-direct {p0, v5}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->showTargets(Z)V
  
      const/16 v0, 0xc8
  
--- 4237,4243 ----
  
      invoke-direct {p0}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->deactivateTargets()V
  
!     invoke-virtual {p0, v5}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->showTargets(Z)V
  
      const/16 v0, 0xc8
  
***************
*** 4280,4286 ****
      goto :goto_0
  
      :pswitch_5
!     invoke-direct {p0}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->doFinish()V
  
      goto :goto_0
  
--- 4280,4286 ----
      goto :goto_0
  
      :pswitch_5
!     invoke-virtual {p0}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->doFinish()V
  
      goto :goto_0
  
***************
*** 4334,4340 ****
      :cond_0
      const/4 v3, 0x2
  
!     invoke-direct {p0, v3, p1, p2}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->switchToState(IFF)V
  
      invoke-direct {p0, v0, v1}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->updateGlowPosition(FF)V
  
--- 4334,4340 ----
      :cond_0
      const/4 v3, 0x2
  
!     invoke-virtual {p0, v3, p1, p2}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->switchToState(IFF)V
  
      invoke-direct {p0, v0, v1}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->updateGlowPosition(FF)V
  
***************
*** 4447,4458 ****
      move-result v0
  
      .local v0, angle:F
!     invoke-direct {p0, p1, p2, p3, v0}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->updateTargetPosition(IFFF)V
  
      return-void
  .end method
  
! .method private updateTargetPosition(IFFF)V
      .locals 6
      .parameter "i"
      .parameter "centerX"
--- 4447,4458 ----
      move-result v0
  
      .local v0, angle:F
!     invoke-virtual {p0, p1, p2, p3, v0}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->updateTargetPosition(IFFF)V
  
      return-void
  .end method
  
! .method updateTargetPosition(IFFF)V
      .locals 6
      .parameter "i"
      .parameter "centerX"
***************
*** 4571,4577 ****
  
      move-result v3
  
!     invoke-direct {p0, v1, p1, p2, v3}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->updateTargetPosition(IFFF)V
  
      :cond_1
      add-int/lit8 v1, v1, 0x1
--- 4571,4577 ----
  
      move-result v3
  
!     invoke-virtual {p0, v1, p1, p2, v3}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->updateTargetPosition(IFFF)V
  
      :cond_1
      add-int/lit8 v1, v1, 0x1
***************
*** 4632,4637 ****
--- 4632,4705 ----
  
  
  # virtual methods
+ .method callAnnounceTargets()V
+     .locals 0
+ 
+     .prologue
+     invoke-direct {p0}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->announceTargets()V
+ 
+     return-void
+ .end method
+ 
+ .method callDeactivateTargets()V
+     .locals 0
+ 
+     .prologue
+     invoke-direct {p0}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->deactivateTargets()V
+ 
+     return-void
+ .end method
+ 
+ .method callDispatchTriggerEvent(I)V
+     .locals 0
+     .parameter "whichTarget"
+ 
+     .prologue
+     invoke-direct {p0, p1}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->dispatchTriggerEvent(I)V
+ 
+     return-void
+ .end method
+ 
+ .method callGetTargetDescription(I)Ljava/lang/String;
+     .locals 1
+     .parameter "index"
+ 
+     .prologue
+     invoke-direct {p0, p1}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->getTargetDescription(I)Ljava/lang/String;
+ 
+     move-result-object v0
+ 
+     return-object v0
+ .end method
+ 
+ .method callSetGrabbedState(I)V
+     .locals 0
+     .parameter "newState"
+ 
+     .prologue
+     invoke-direct {p0, p1}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->setGrabbedState(I)V
+ 
+     return-void
+ .end method
+ 
+ .method callVibrate()V
+     .locals 0
+ 
+     .prologue
+     invoke-direct {p0}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->vibrate()V
+ 
+     return-void
+ .end method
+ 
+ .method getActiveTarget()I
+     .locals 1
+ 
+     .prologue
+     iget v0, p0, Lcom/android/internal/widget/multiwaveview/GlowPadView;->mActiveTarget:I
+ 
+     return v0
+ .end method
+ 
  .method public getDirectionDescriptionsResourceId()I
      .locals 1
  
***************
*** 4641,4646 ****
--- 4709,4768 ----
      return v0
  .end method
  
+ .method getDragging()Z
+     .locals 1
+ 
+     .prologue
+     iget-boolean v0, p0, Lcom/android/internal/widget/multiwaveview/GlowPadView;->mDragging:Z
+ 
+     return v0
+ .end method
+ 
+ .method getHandleDrawable()Lcom/android/internal/widget/multiwaveview/TargetDrawable;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/widget/multiwaveview/GlowPadView;->mHandleDrawable:Lcom/android/internal/widget/multiwaveview/TargetDrawable;
+ 
+     return-object v0
+ .end method
+ 
+ .method getOuterRadius()F
+     .locals 1
+ 
+     .prologue
+     iget v0, p0, Lcom/android/internal/widget/multiwaveview/GlowPadView;->mOuterRadius:F
+ 
+     return v0
+ .end method
+ 
+ .method getOuterRing()Lcom/android/internal/widget/multiwaveview/TargetDrawable;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/widget/multiwaveview/GlowPadView;->mOuterRing:Lcom/android/internal/widget/multiwaveview/TargetDrawable;
+ 
+     return-object v0
+ .end method
+ 
+ .method getResetListener()Landroid/animation/Animator$AnimatorListener;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/widget/multiwaveview/GlowPadView;->mResetListener:Landroid/animation/Animator$AnimatorListener;
+ 
+     return-object v0
+ .end method
+ 
+ .method getResetListenerWithPing()Landroid/animation/Animator$AnimatorListener;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/widget/multiwaveview/GlowPadView;->mResetListenerWithPing:Landroid/animation/Animator$AnimatorListener;
+ 
+     return-object v0
+ .end method
+ 
  .method public getResourceIdForTarget(I)I
      .locals 2
      .parameter "index"
***************
*** 4744,4749 ****
--- 4866,4880 ----
      return v0
  .end method
  
+ .method getSnapMargin()F
+     .locals 1
+ 
+     .prologue
+     iget v0, p0, Lcom/android/internal/widget/multiwaveview/GlowPadView;->mSnapMargin:F
+ 
+     return v0
+ .end method
+ 
  .method protected getSuggestedMinimumHeight()I
      .locals 3
  
***************
*** 4810,4815 ****
--- 4941,4955 ----
      return v0
  .end method
  
+ .method getTargetAnimations()Lcom/android/internal/widget/multiwaveview/GlowPadView$AnimationBundle;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/widget/multiwaveview/GlowPadView;->mTargetAnimations:Lcom/android/internal/widget/multiwaveview/GlowPadView$AnimationBundle;
+ 
+     return-object v0
+ .end method
+ 
  .method public getTargetDescriptionsResourceId()I
      .locals 1
  
***************
*** 4819,4824 ****
--- 4959,4982 ----
      return v0
  .end method
  
+ .method getTargetDrawables()Ljava/util/ArrayList;
+     .locals 1
+     .annotation system Ldalvik/annotation/Signature;
+         value = {
+             "()",
+             "Ljava/util/ArrayList",
+             "<",
+             "Lcom/android/internal/widget/multiwaveview/TargetDrawable;",
+             ">;"
+         }
+     .end annotation
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/widget/multiwaveview/GlowPadView;->mTargetDrawables:Ljava/util/ArrayList;
+ 
+     return-object v0
+ .end method
+ 
  .method public getTargetPosition(I)I
      .locals 3
      .parameter "resourceId"
***************
*** 4879,4884 ****
--- 5037,5069 ----
      return v0
  .end method
  
+ .method getUpdateListener()Landroid/animation/ValueAnimator$AnimatorUpdateListener;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/widget/multiwaveview/GlowPadView;->mUpdateListener:Landroid/animation/ValueAnimator$AnimatorUpdateListener;
+ 
+     return-object v0
+ .end method
+ 
+ .method getWaveCenterX()F
+     .locals 1
+ 
+     .prologue
+     iget v0, p0, Lcom/android/internal/widget/multiwaveview/GlowPadView;->mWaveCenterX:F
+ 
+     return v0
+ .end method
+ 
+ .method getWaveCenterY()F
+     .locals 1
+ 
+     .prologue
+     iget v0, p0, Lcom/android/internal/widget/multiwaveview/GlowPadView;->mWaveCenterY:F
+ 
+     return v0
+ .end method
+ 
  .method protected onDraw(Landroid/graphics/Canvas;)V
      .locals 4
      .parameter "canvas"
***************
*** 5082,5088 ****
  
      const/4 v7, 0x0
  
!     invoke-direct {p0, v6, v7}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->hideTargets(ZZ)V
  
      const/4 v6, 0x0
  
--- 5267,5273 ----
  
      const/4 v7, 0x0
  
!     invoke-virtual {p0, v6, v7}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->hideTargets(ZZ)V
  
      const/4 v6, 0x0
  
***************
*** 5206,5226 ****
      :pswitch_1
      invoke-direct {p0, p1}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->handleDown(Landroid/view/MotionEvent;)V
  
!     invoke-direct {p0, p1}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->handleMove(Landroid/view/MotionEvent;)V
  
      const/4 v1, 0x1
  
      goto :goto_0
  
      :pswitch_2
!     invoke-direct {p0, p1}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->handleMove(Landroid/view/MotionEvent;)V
  
      const/4 v1, 0x1
  
      goto :goto_0
  
      :pswitch_3
!     invoke-direct {p0, p1}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->handleMove(Landroid/view/MotionEvent;)V
  
      invoke-direct {p0, p1}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->handleUp(Landroid/view/MotionEvent;)V
  
--- 5391,5411 ----
      :pswitch_1
      invoke-direct {p0, p1}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->handleDown(Landroid/view/MotionEvent;)V
  
!     invoke-virtual {p0, p1}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->handleMove(Landroid/view/MotionEvent;)V
  
      const/4 v1, 0x1
  
      goto :goto_0
  
      :pswitch_2
!     invoke-virtual {p0, p1}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->handleMove(Landroid/view/MotionEvent;)V
  
      const/4 v1, 0x1
  
      goto :goto_0
  
      :pswitch_3
!     invoke-virtual {p0, p1}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->handleMove(Landroid/view/MotionEvent;)V
  
      invoke-direct {p0, p1}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->handleUp(Landroid/view/MotionEvent;)V
  
***************
*** 5229,5235 ****
      goto :goto_0
  
      :pswitch_4
!     invoke-direct {p0, p1}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->handleMove(Landroid/view/MotionEvent;)V
  
      invoke-direct {p0, p1}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->handleCancel(Landroid/view/MotionEvent;)V
  
--- 5414,5420 ----
      goto :goto_0
  
      :pswitch_4
!     invoke-virtual {p0, p1}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->handleMove(Landroid/view/MotionEvent;)V
  
      invoke-direct {p0, p1}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->handleCancel(Landroid/view/MotionEvent;)V
  
***************
*** 5497,5503 ****
  
      invoke-direct {p0}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->stopAndHideWaveAnimation()V
  
!     invoke-direct {p0, p1, v1}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->hideTargets(ZZ)V
  
      const/4 v0, 0x0
  
--- 5682,5688 ----
  
      invoke-direct {p0}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->stopAndHideWaveAnimation()V
  
!     invoke-virtual {p0, p1, v1}, Lcom/android/internal/widget/multiwaveview/GlowPadView;->hideTargets(ZZ)V
  
      const/4 v0, 0x0
  
***************
*** 5541,5546 ****
--- 5726,5751 ----
      return-void
  .end method
  
+ .method setActiveTarget(I)V
+     .locals 0
+     .parameter "value"
+ 
+     .prologue
+     iput p1, p0, Lcom/android/internal/widget/multiwaveview/GlowPadView;->mActiveTarget:I
+ 
+     return-void
+ .end method
+ 
+ .method setAnimatingTargets(Z)V
+     .locals 0
+     .parameter "value"
+ 
+     .prologue
+     iput-boolean p1, p0, Lcom/android/internal/widget/multiwaveview/GlowPadView;->mAnimatingTargets:Z
+ 
+     return-void
+ .end method
+ 
  .method public setDirectionDescriptionsResourceId(I)V
      .locals 1
      .parameter "resourceId"
***************
*** 5560,5565 ****
--- 5765,5780 ----
      return-void
  .end method
  
+ .method setDragging(Z)V
+     .locals 0
+     .parameter "value"
+ 
+     .prologue
+     iput-boolean p1, p0, Lcom/android/internal/widget/multiwaveview/GlowPadView;->mDragging:Z
+ 
+     return-void
+ .end method
+ 
  .method public setEnableTarget(IZ)V
      .locals 3
      .parameter "resourceId"
***************
*** 5606,5611 ****
--- 5821,5836 ----
      goto :goto_0
  .end method
  
+ .method setGlowRadius(F)V
+     .locals 0
+     .parameter "value"
+ 
+     .prologue
+     iput p1, p0, Lcom/android/internal/widget/multiwaveview/GlowPadView;->mGlowRadius:F
+ 
+     return-void
+ .end method
+ 
  .method public setOnTriggerListener(Lcom/android/internal/widget/multiwaveview/GlowPadView$OnTriggerListener;)V
      .locals 0
      .parameter "listener"
